{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A7J11_01233+.assembled</th>\n",
       "      <th>A7J11_01233+.ref_seq</th>\n",
       "      <th>A7J11_01233+.pct_id</th>\n",
       "      <th>A7J11_01233+.ctg_cov</th>\n",
       "      <th>aac_3__II-.assembled</th>\n",
       "      <th>aac_3__II-.ref_seq</th>\n",
       "      <th>aac_3__II-.pct_id</th>\n",
       "      <th>aac_3__II-.ctg_cov</th>\n",
       "      <th>aac_6___Ib+.assembled</th>\n",
       "      <th>aac_6___Ib+.ref_seq</th>\n",
       "      <th>...</th>\n",
       "      <th>ompK36.N304E</th>\n",
       "      <th>ompK36.N49S</th>\n",
       "      <th>ompK36.S346N</th>\n",
       "      <th>ompK36.T222L</th>\n",
       "      <th>ompK37.assembled</th>\n",
       "      <th>ompK37.E244D</th>\n",
       "      <th>ompK37.I128M</th>\n",
       "      <th>ompK37.I70M</th>\n",
       "      <th>ompK37.N230G</th>\n",
       "      <th>ramR.assembled</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KPN12_S2_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>99.83</td>\n",
       "      <td>51.8</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>aac_6___Ib.NG_056043.1</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN13_S3_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>99.65</td>\n",
       "      <td>37.7</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>aac_6___Ib4.NG_051844.1</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN14_S4_L001</th>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN15_S5_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>99.83</td>\n",
       "      <td>82.3</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>aac_3__IId.NG_047251.1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>98.8</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>aac_6___Ib.NG_056043.1</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNH_S7_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>99.83</td>\n",
       "      <td>35.9</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>aac_6___Ib.NG_056043.1</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNI_S8_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>99.65</td>\n",
       "      <td>30.3</td>\n",
       "      <td>partial</td>\n",
       "      <td>aac_3__IId.NG_047251.1</td>\n",
       "      <td>99.76</td>\n",
       "      <td>2.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>aac_6___Ib4.NG_051844.1</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNJ_S2_L001</th>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNK_S1_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>99.83</td>\n",
       "      <td>111.8</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>aac_3__IId.NG_047251.1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>151.5</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>aac_6___Ib.NG_056043.1</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 232 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              A7J11_01233+.assembled     A7J11_01233+.ref_seq  \\\n",
       "name                                                            \n",
       "KPN12_S2_L001                partial  A7J11_01233.NG_047213.1   \n",
       "KPN13_S3_L001                partial  A7J11_01233.NG_047213.1   \n",
       "KPN14_S4_L001                     no                      NaN   \n",
       "KPN15_S5_L001                partial  A7J11_01233.NG_047213.1   \n",
       "KPNH_S7_L001                 partial  A7J11_01233.NG_047213.1   \n",
       "KPNI_S8_L001                 partial  A7J11_01233.NG_047213.1   \n",
       "KPNJ_S2_L001                      no                      NaN   \n",
       "KPNK_S1_L001                 partial  A7J11_01233.NG_047213.1   \n",
       "\n",
       "               A7J11_01233+.pct_id  A7J11_01233+.ctg_cov aac_3__II-.assembled  \\\n",
       "name                                                                            \n",
       "KPN12_S2_L001                99.83                  51.8                   no   \n",
       "KPN13_S3_L001                99.65                  37.7                   no   \n",
       "KPN14_S4_L001                  NaN                   NaN                   no   \n",
       "KPN15_S5_L001                99.83                  82.3        yes_nonunique   \n",
       "KPNH_S7_L001                 99.83                  35.9                   no   \n",
       "KPNI_S8_L001                 99.65                  30.3              partial   \n",
       "KPNJ_S2_L001                   NaN                   NaN                   no   \n",
       "KPNK_S1_L001                 99.83                 111.8        yes_nonunique   \n",
       "\n",
       "                   aac_3__II-.ref_seq  aac_3__II-.pct_id  aac_3__II-.ctg_cov  \\\n",
       "name                                                                           \n",
       "KPN12_S2_L001                     NaN                NaN                 NaN   \n",
       "KPN13_S3_L001                     NaN                NaN                 NaN   \n",
       "KPN14_S4_L001                     NaN                NaN                 NaN   \n",
       "KPN15_S5_L001  aac_3__IId.NG_047251.1             100.00                98.8   \n",
       "KPNH_S7_L001                      NaN                NaN                 NaN   \n",
       "KPNI_S8_L001   aac_3__IId.NG_047251.1              99.76                 2.0   \n",
       "KPNJ_S2_L001                      NaN                NaN                 NaN   \n",
       "KPNK_S1_L001   aac_3__IId.NG_047251.1             100.00               151.5   \n",
       "\n",
       "              aac_6___Ib+.assembled      aac_6___Ib+.ref_seq  ...  \\\n",
       "name                                                          ...   \n",
       "KPN12_S2_L001                   yes   aac_6___Ib.NG_056043.1  ...   \n",
       "KPN13_S3_L001                   yes  aac_6___Ib4.NG_051844.1  ...   \n",
       "KPN14_S4_L001                    no                      NaN  ...   \n",
       "KPN15_S5_L001         yes_nonunique   aac_6___Ib.NG_056043.1  ...   \n",
       "KPNH_S7_L001                    yes   aac_6___Ib.NG_056043.1  ...   \n",
       "KPNI_S8_L001                    yes  aac_6___Ib4.NG_051844.1  ...   \n",
       "KPNJ_S2_L001                     no                      NaN  ...   \n",
       "KPNK_S1_L001          yes_nonunique   aac_6___Ib.NG_056043.1  ...   \n",
       "\n",
       "               ompK36.N304E  ompK36.N49S ompK36.S346N ompK36.T222L  \\\n",
       "name                                                                 \n",
       "KPN12_S2_L001           yes          yes          yes           no   \n",
       "KPN13_S3_L001            no          yes           no           no   \n",
       "KPN14_S4_L001           yes          yes          yes          yes   \n",
       "KPN15_S5_L001            no          yes          yes           no   \n",
       "KPNH_S7_L001            yes          yes          yes          yes   \n",
       "KPNI_S8_L001             no          yes           no           no   \n",
       "KPNJ_S2_L001            yes          yes          yes          yes   \n",
       "KPNK_S1_L001             no          yes           no           no   \n",
       "\n",
       "               ompK37.assembled  ompK37.E244D ompK37.I128M ompK37.I70M  \\\n",
       "name                                                                     \n",
       "KPN12_S2_L001               yes           yes          yes         yes   \n",
       "KPN13_S3_L001               yes            no          yes         yes   \n",
       "KPN14_S4_L001               yes            no          yes         yes   \n",
       "KPN15_S5_L001               yes            no          yes         yes   \n",
       "KPNH_S7_L001                yes           yes          yes         yes   \n",
       "KPNI_S8_L001                yes            no          yes         yes   \n",
       "KPNJ_S2_L001                yes            no          yes         yes   \n",
       "KPNK_S1_L001                yes            no          yes         yes   \n",
       "\n",
       "               ompK37.N230G  ramR.assembled  \n",
       "name                                         \n",
       "KPN12_S2_L001           yes             yes  \n",
       "KPN13_S3_L001            no     interrupted  \n",
       "KPN14_S4_L001            no     interrupted  \n",
       "KPN15_S5_L001            no     interrupted  \n",
       "KPNH_S7_L001            yes             yes  \n",
       "KPNI_S8_L001             no     interrupted  \n",
       "KPNJ_S2_L001             no     interrupted  \n",
       "KPNK_S1_L001             no     interrupted  \n",
       "\n",
       "[8 rows x 232 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "datos = pd.read_csv(\"combined_summary_k_pneumoniae.csv\", sep=\";\", index_col=0)\n",
    "datos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A7J11_01233+.assembled</th>\n",
       "      <th>A7J11_01233+.ref_seq</th>\n",
       "      <th>A7J11_01233+.pct_id</th>\n",
       "      <th>A7J11_01233+.ctg_cov</th>\n",
       "      <th>aac_3__II-.assembled</th>\n",
       "      <th>aac_3__II-.ref_seq</th>\n",
       "      <th>aac_3__II-.pct_id</th>\n",
       "      <th>aac_3__II-.ctg_cov</th>\n",
       "      <th>aac_6___Ib+.assembled</th>\n",
       "      <th>aac_6___Ib+.ref_seq</th>\n",
       "      <th>...</th>\n",
       "      <th>ompK36.N304E</th>\n",
       "      <th>ompK36.N49S</th>\n",
       "      <th>ompK36.S346N</th>\n",
       "      <th>ompK36.T222L</th>\n",
       "      <th>ompK37.assembled</th>\n",
       "      <th>ompK37.E244D</th>\n",
       "      <th>ompK37.I128M</th>\n",
       "      <th>ompK37.I70M</th>\n",
       "      <th>ompK37.N230G</th>\n",
       "      <th>ramR.assembled</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KPN12_S2_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>99.83</td>\n",
       "      <td>51.8</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>aac_6___Ib.NG_056043.1</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN13_S3_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>99.65</td>\n",
       "      <td>37.7</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>aac_6___Ib4.NG_051844.1</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN14_S4_L001</th>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN15_S5_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>99.83</td>\n",
       "      <td>82.3</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>aac_3__IId.NG_047251.1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>98.8</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>aac_6___Ib.NG_056043.1</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNH_S7_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>99.83</td>\n",
       "      <td>35.9</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>aac_6___Ib.NG_056043.1</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNI_S8_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>99.65</td>\n",
       "      <td>30.3</td>\n",
       "      <td>partial</td>\n",
       "      <td>aac_3__IId.NG_047251.1</td>\n",
       "      <td>99.76</td>\n",
       "      <td>2.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>aac_6___Ib4.NG_051844.1</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNJ_S2_L001</th>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNK_S1_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>99.83</td>\n",
       "      <td>111.8</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>aac_3__IId.NG_047251.1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>151.5</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>aac_6___Ib.NG_056043.1</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 232 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              A7J11_01233+.assembled     A7J11_01233+.ref_seq  \\\n",
       "samples                                                         \n",
       "KPN12_S2_L001                partial  A7J11_01233.NG_047213.1   \n",
       "KPN13_S3_L001                partial  A7J11_01233.NG_047213.1   \n",
       "KPN14_S4_L001                     no                      NaN   \n",
       "KPN15_S5_L001                partial  A7J11_01233.NG_047213.1   \n",
       "KPNH_S7_L001                 partial  A7J11_01233.NG_047213.1   \n",
       "KPNI_S8_L001                 partial  A7J11_01233.NG_047213.1   \n",
       "KPNJ_S2_L001                      no                      NaN   \n",
       "KPNK_S1_L001                 partial  A7J11_01233.NG_047213.1   \n",
       "\n",
       "               A7J11_01233+.pct_id  A7J11_01233+.ctg_cov aac_3__II-.assembled  \\\n",
       "samples                                                                         \n",
       "KPN12_S2_L001                99.83                  51.8                   no   \n",
       "KPN13_S3_L001                99.65                  37.7                   no   \n",
       "KPN14_S4_L001                  NaN                   NaN                   no   \n",
       "KPN15_S5_L001                99.83                  82.3        yes_nonunique   \n",
       "KPNH_S7_L001                 99.83                  35.9                   no   \n",
       "KPNI_S8_L001                 99.65                  30.3              partial   \n",
       "KPNJ_S2_L001                   NaN                   NaN                   no   \n",
       "KPNK_S1_L001                 99.83                 111.8        yes_nonunique   \n",
       "\n",
       "                   aac_3__II-.ref_seq  aac_3__II-.pct_id  aac_3__II-.ctg_cov  \\\n",
       "samples                                                                        \n",
       "KPN12_S2_L001                     NaN                NaN                 NaN   \n",
       "KPN13_S3_L001                     NaN                NaN                 NaN   \n",
       "KPN14_S4_L001                     NaN                NaN                 NaN   \n",
       "KPN15_S5_L001  aac_3__IId.NG_047251.1             100.00                98.8   \n",
       "KPNH_S7_L001                      NaN                NaN                 NaN   \n",
       "KPNI_S8_L001   aac_3__IId.NG_047251.1              99.76                 2.0   \n",
       "KPNJ_S2_L001                      NaN                NaN                 NaN   \n",
       "KPNK_S1_L001   aac_3__IId.NG_047251.1             100.00               151.5   \n",
       "\n",
       "              aac_6___Ib+.assembled      aac_6___Ib+.ref_seq  ...  \\\n",
       "samples                                                       ...   \n",
       "KPN12_S2_L001                   yes   aac_6___Ib.NG_056043.1  ...   \n",
       "KPN13_S3_L001                   yes  aac_6___Ib4.NG_051844.1  ...   \n",
       "KPN14_S4_L001                    no                      NaN  ...   \n",
       "KPN15_S5_L001         yes_nonunique   aac_6___Ib.NG_056043.1  ...   \n",
       "KPNH_S7_L001                    yes   aac_6___Ib.NG_056043.1  ...   \n",
       "KPNI_S8_L001                    yes  aac_6___Ib4.NG_051844.1  ...   \n",
       "KPNJ_S2_L001                     no                      NaN  ...   \n",
       "KPNK_S1_L001          yes_nonunique   aac_6___Ib.NG_056043.1  ...   \n",
       "\n",
       "               ompK36.N304E  ompK36.N49S ompK36.S346N ompK36.T222L  \\\n",
       "samples                                                              \n",
       "KPN12_S2_L001           yes          yes          yes           no   \n",
       "KPN13_S3_L001            no          yes           no           no   \n",
       "KPN14_S4_L001           yes          yes          yes          yes   \n",
       "KPN15_S5_L001            no          yes          yes           no   \n",
       "KPNH_S7_L001            yes          yes          yes          yes   \n",
       "KPNI_S8_L001             no          yes           no           no   \n",
       "KPNJ_S2_L001            yes          yes          yes          yes   \n",
       "KPNK_S1_L001             no          yes           no           no   \n",
       "\n",
       "               ompK37.assembled  ompK37.E244D ompK37.I128M ompK37.I70M  \\\n",
       "samples                                                                  \n",
       "KPN12_S2_L001               yes           yes          yes         yes   \n",
       "KPN13_S3_L001               yes            no          yes         yes   \n",
       "KPN14_S4_L001               yes            no          yes         yes   \n",
       "KPN15_S5_L001               yes            no          yes         yes   \n",
       "KPNH_S7_L001                yes           yes          yes         yes   \n",
       "KPNI_S8_L001                yes            no          yes         yes   \n",
       "KPNJ_S2_L001                yes            no          yes         yes   \n",
       "KPNK_S1_L001                yes            no          yes         yes   \n",
       "\n",
       "               ompK37.N230G  ramR.assembled  \n",
       "samples                                      \n",
       "KPN12_S2_L001           yes             yes  \n",
       "KPN13_S3_L001            no     interrupted  \n",
       "KPN14_S4_L001            no     interrupted  \n",
       "KPN15_S5_L001            no     interrupted  \n",
       "KPNH_S7_L001            yes             yes  \n",
       "KPNI_S8_L001             no     interrupted  \n",
       "KPNJ_S2_L001             no     interrupted  \n",
       "KPNK_S1_L001             no     interrupted  \n",
       "\n",
       "[8 rows x 232 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "datos=datos.rename_axis('samples')\n",
    "datos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>known_variants</th>\n",
       "      <th>acrR.F172S</th>\n",
       "      <th>acrR.F197I</th>\n",
       "      <th>acrR.G164A</th>\n",
       "      <th>acrR.K201M</th>\n",
       "      <th>acrR.L195V</th>\n",
       "      <th>acrR.P161R</th>\n",
       "      <th>acrR.R173G</th>\n",
       "      <th>gyrA.D87G</th>\n",
       "      <th>ompK36.A183P</th>\n",
       "      <th>...</th>\n",
       "      <th>ompK36.L59V</th>\n",
       "      <th>ompK36.N218H</th>\n",
       "      <th>ompK36.N304E</th>\n",
       "      <th>ompK36.N49S</th>\n",
       "      <th>ompK36.S346N</th>\n",
       "      <th>ompK36.T222L</th>\n",
       "      <th>ompK37.E244D</th>\n",
       "      <th>ompK37.I128M</th>\n",
       "      <th>ompK37.I70M</th>\n",
       "      <th>ompK37.N230G</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KPN12_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN13_S3_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN14_S4_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN15_S5_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNH_S7_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNI_S8_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNJ_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNK_S1_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               known_variants acrR.F172S acrR.F197I acrR.G164A acrR.K201M  \\\n",
       "samples                                                                     \n",
       "KPN12_S2_L001             NaN        yes        yes        yes        yes   \n",
       "KPN13_S3_L001             NaN        yes        yes        yes        yes   \n",
       "KPN14_S4_L001             NaN        yes        yes        yes        yes   \n",
       "KPN15_S5_L001             NaN        yes        yes        yes        yes   \n",
       "KPNH_S7_L001              NaN        yes        yes        yes        yes   \n",
       "KPNI_S8_L001              NaN        yes        yes        yes        yes   \n",
       "KPNJ_S2_L001              NaN        yes        yes        yes        yes   \n",
       "KPNK_S1_L001              NaN        yes        yes        yes        yes   \n",
       "\n",
       "              acrR.L195V acrR.P161R acrR.R173G gyrA.D87G ompK36.A183P  ...  \\\n",
       "samples                                                                ...   \n",
       "KPN12_S2_L001        yes        yes        yes        no           no  ...   \n",
       "KPN13_S3_L001        yes        yes        yes        no          yes  ...   \n",
       "KPN14_S4_L001        yes        yes        yes        no           no  ...   \n",
       "KPN15_S5_L001        yes        yes        yes       NaN           no  ...   \n",
       "KPNH_S7_L001         yes        yes        yes        no           no  ...   \n",
       "KPNI_S8_L001         yes        yes        yes        no          yes  ...   \n",
       "KPNJ_S2_L001         yes        yes        yes        no           no  ...   \n",
       "KPNK_S1_L001         yes        yes        yes       yes           no  ...   \n",
       "\n",
       "              ompK36.L59V ompK36.N218H ompK36.N304E ompK36.N49S ompK36.S346N  \\\n",
       "samples                                                                        \n",
       "KPN12_S2_L001         yes           no          yes         yes          yes   \n",
       "KPN13_S3_L001         yes          yes           no         yes           no   \n",
       "KPN14_S4_L001         yes           no          yes         yes          yes   \n",
       "KPN15_S5_L001         yes          yes           no         yes          yes   \n",
       "KPNH_S7_L001          yes           no          yes         yes          yes   \n",
       "KPNI_S8_L001          yes          yes           no         yes           no   \n",
       "KPNJ_S2_L001          yes           no          yes         yes          yes   \n",
       "KPNK_S1_L001          yes          yes           no         yes           no   \n",
       "\n",
       "              ompK36.T222L ompK37.E244D ompK37.I128M ompK37.I70M ompK37.N230G  \n",
       "samples                                                                        \n",
       "KPN12_S2_L001           no          yes          yes         yes          yes  \n",
       "KPN13_S3_L001           no           no          yes         yes           no  \n",
       "KPN14_S4_L001          yes           no          yes         yes           no  \n",
       "KPN15_S5_L001           no           no          yes         yes           no  \n",
       "KPNH_S7_L001           yes          yes          yes         yes          yes  \n",
       "KPNI_S8_L001            no           no          yes         yes           no  \n",
       "KPNJ_S2_L001           yes           no          yes         yes           no  \n",
       "KPNK_S1_L001            no           no          yes         yes           no  \n",
       "\n",
       "[8 rows x 31 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#####------------------>>>>>>> mutaciones de genes\n",
    "\n",
    "df_assembled = datos[datos.columns.drop(list(datos.filter(regex='.assembled')))]\n",
    "df_ref_seq = df_assembled[df_assembled.columns.drop(list(df_assembled.filter(regex='.ref_seq')))]\n",
    "df_pct_id = df_ref_seq[df_ref_seq.columns.drop(list(df_ref_seq.filter(regex='.pct_id')))]\n",
    "mutations = df_pct_id[df_pct_id.columns.drop(list(df_pct_id.filter(regex='.ctg_cov')))]\n",
    "mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/felipedb/opt/anaconda3/envs/agrosavia/lib/python3.7/site-packages/pandas/core/ops/array_ops.py:253: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  res_values = method(rvalues)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>known_variants</th>\n",
       "      <th>acrR.F172S</th>\n",
       "      <th>acrR.F197I</th>\n",
       "      <th>acrR.G164A</th>\n",
       "      <th>acrR.K201M</th>\n",
       "      <th>acrR.L195V</th>\n",
       "      <th>acrR.P161R</th>\n",
       "      <th>acrR.R173G</th>\n",
       "      <th>gyrA.D87G</th>\n",
       "      <th>ompK36.A183P</th>\n",
       "      <th>...</th>\n",
       "      <th>ompK36.L59V</th>\n",
       "      <th>ompK36.N218H</th>\n",
       "      <th>ompK36.N304E</th>\n",
       "      <th>ompK36.N49S</th>\n",
       "      <th>ompK36.S346N</th>\n",
       "      <th>ompK36.T222L</th>\n",
       "      <th>ompK37.E244D</th>\n",
       "      <th>ompK37.I128M</th>\n",
       "      <th>ompK37.I70M</th>\n",
       "      <th>ompK37.N230G</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KPN12_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN13_S3_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN14_S4_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN15_S5_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNH_S7_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNI_S8_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNJ_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNK_S1_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               known_variants acrR.F172S acrR.F197I acrR.G164A acrR.K201M  \\\n",
       "samples                                                                     \n",
       "KPN12_S2_L001             NaN        yes        yes        yes        yes   \n",
       "KPN13_S3_L001             NaN        yes        yes        yes        yes   \n",
       "KPN14_S4_L001             NaN        yes        yes        yes        yes   \n",
       "KPN15_S5_L001             NaN        yes        yes        yes        yes   \n",
       "KPNH_S7_L001              NaN        yes        yes        yes        yes   \n",
       "KPNI_S8_L001              NaN        yes        yes        yes        yes   \n",
       "KPNJ_S2_L001              NaN        yes        yes        yes        yes   \n",
       "KPNK_S1_L001              NaN        yes        yes        yes        yes   \n",
       "\n",
       "              acrR.L195V acrR.P161R acrR.R173G gyrA.D87G ompK36.A183P  ...  \\\n",
       "samples                                                                ...   \n",
       "KPN12_S2_L001        yes        yes        yes       NaN          NaN  ...   \n",
       "KPN13_S3_L001        yes        yes        yes       NaN          yes  ...   \n",
       "KPN14_S4_L001        yes        yes        yes       NaN          NaN  ...   \n",
       "KPN15_S5_L001        yes        yes        yes       NaN          NaN  ...   \n",
       "KPNH_S7_L001         yes        yes        yes       NaN          NaN  ...   \n",
       "KPNI_S8_L001         yes        yes        yes       NaN          yes  ...   \n",
       "KPNJ_S2_L001         yes        yes        yes       NaN          NaN  ...   \n",
       "KPNK_S1_L001         yes        yes        yes       yes          NaN  ...   \n",
       "\n",
       "              ompK36.L59V ompK36.N218H ompK36.N304E ompK36.N49S ompK36.S346N  \\\n",
       "samples                                                                        \n",
       "KPN12_S2_L001         yes          NaN          yes         yes          yes   \n",
       "KPN13_S3_L001         yes          yes          NaN         yes          NaN   \n",
       "KPN14_S4_L001         yes          NaN          yes         yes          yes   \n",
       "KPN15_S5_L001         yes          yes          NaN         yes          yes   \n",
       "KPNH_S7_L001          yes          NaN          yes         yes          yes   \n",
       "KPNI_S8_L001          yes          yes          NaN         yes          NaN   \n",
       "KPNJ_S2_L001          yes          NaN          yes         yes          yes   \n",
       "KPNK_S1_L001          yes          yes          NaN         yes          NaN   \n",
       "\n",
       "              ompK36.T222L ompK37.E244D ompK37.I128M ompK37.I70M ompK37.N230G  \n",
       "samples                                                                        \n",
       "KPN12_S2_L001          NaN          yes          yes         yes          yes  \n",
       "KPN13_S3_L001          NaN          NaN          yes         yes          NaN  \n",
       "KPN14_S4_L001          yes          NaN          yes         yes          NaN  \n",
       "KPN15_S5_L001          NaN          NaN          yes         yes          NaN  \n",
       "KPNH_S7_L001           yes          yes          yes         yes          yes  \n",
       "KPNI_S8_L001           NaN          NaN          yes         yes          NaN  \n",
       "KPNJ_S2_L001           yes          NaN          yes         yes          NaN  \n",
       "KPNK_S1_L001           NaN          NaN          yes         yes          NaN  \n",
       "\n",
       "[8 rows x 31 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutations_yes=mutations[mutations==\"yes\"]\n",
    "mutations_yes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "samples                    \n",
       "KPN12_S2_L001  acrR.F172S      yes\n",
       "               acrR.F197I      yes\n",
       "               acrR.G164A      yes\n",
       "               acrR.K201M      yes\n",
       "               acrR.L195V      yes\n",
       "                              ... \n",
       "KPNK_S1_L001   ompK36.L59V     yes\n",
       "               ompK36.N218H    yes\n",
       "               ompK36.N49S     yes\n",
       "               ompK37.I128M    yes\n",
       "               ompK37.I70M     yes\n",
       "Length: 153, dtype: object"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutations_st=mutations_yes.stack()\n",
    "mutations_st "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>mutation_result</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPN12_S2_L001</th>\n",
       "      <th>acrR.F172S</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR.F197I</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR.G164A</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR.K201M</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR.L195V</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPNK_S1_L001</th>\n",
       "      <th>ompK36.L59V</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK36.N218H</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK36.N49S</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK37.I128M</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK37.I70M</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           mutation_result\n",
       "samples                                   \n",
       "KPN12_S2_L001 acrR.F172S               yes\n",
       "              acrR.F197I               yes\n",
       "              acrR.G164A               yes\n",
       "              acrR.K201M               yes\n",
       "              acrR.L195V               yes\n",
       "...                                    ...\n",
       "KPNK_S1_L001  ompK36.L59V              yes\n",
       "              ompK36.N218H             yes\n",
       "              ompK36.N49S              yes\n",
       "              ompK37.I128M             yes\n",
       "              ompK37.I70M              yes\n",
       "\n",
       "[153 rows x 1 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#####------------------->>>>>>>> mutation dataframe final result\n",
    "mutations_yes=pd.DataFrame(mutations_st)\n",
    "mutation_result_df=mutations_yes.rename(columns = {0:'mutation_result'}, inplace = False)\n",
    "mutation_result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th>mutation_result</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>acrR_F172S</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>acrR_F197I</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>acrR_G164A</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>acrR_K201M</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>acrR_L195V</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>ompK36_L59V</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>ompK36_N218H</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>ompK36_N49S</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>ompK37_I128M</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>ompK37_I70M</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           samples       level_1 mutation_result\n",
       "0    KPN12_S2_L001    acrR_F172S             yes\n",
       "1    KPN12_S2_L001    acrR_F197I             yes\n",
       "2    KPN12_S2_L001    acrR_G164A             yes\n",
       "3    KPN12_S2_L001    acrR_K201M             yes\n",
       "4    KPN12_S2_L001    acrR_L195V             yes\n",
       "..             ...           ...             ...\n",
       "148   KPNK_S1_L001   ompK36_L59V             yes\n",
       "149   KPNK_S1_L001  ompK36_N218H             yes\n",
       "150   KPNK_S1_L001   ompK36_N49S             yes\n",
       "151   KPNK_S1_L001  ompK37_I128M             yes\n",
       "152   KPNK_S1_L001   ompK37_I70M             yes\n",
       "\n",
       "[153 rows x 3 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutation_result_df = mutation_result_df.reset_index(level=[0,1])\n",
    "mutation_result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "mutation_result_df=mutation_result_df.stack().str.replace('.','_').unstack()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>mutation_result</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPN12_S2_L001</th>\n",
       "      <th>acrR_F172S</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_F197I</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_G164A</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_K201M</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_L195V</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPNK_S1_L001</th>\n",
       "      <th>ompK36_L59V</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK36_N218H</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK36_N49S</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK37_I128M</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK37_I70M</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           mutation_result\n",
       "samples       level_1                     \n",
       "KPN12_S2_L001 acrR_F172S               yes\n",
       "              acrR_F197I               yes\n",
       "              acrR_G164A               yes\n",
       "              acrR_K201M               yes\n",
       "              acrR_L195V               yes\n",
       "...                                    ...\n",
       "KPNK_S1_L001  ompK36_L59V              yes\n",
       "              ompK36_N218H             yes\n",
       "              ompK36_N49S              yes\n",
       "              ompK37_I128M             yes\n",
       "              ompK37_I70M              yes\n",
       "\n",
       "[153 rows x 1 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutation_result_df=mutation_result_df.set_index(['samples', 'level_1'])\n",
    "mutation_result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "####----------------------------------->>> .assembled process\n",
    "\n",
    "assembled=datos.columns.str.find(\".assembled\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False, False,  True,\n",
       "       False, False, False, False, False, False, False,  True, False,\n",
       "        True, False, False, False, False, False, False, False, False,\n",
       "       False, False, False, False, False, False, False, False, False,\n",
       "       False,  True, False, False, False, False,  True])"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AS=assembled != -1\n",
    "AS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A7J11_01233+.assembled</th>\n",
       "      <th>aac_3__II-.assembled</th>\n",
       "      <th>aac_6___Ib+.assembled</th>\n",
       "      <th>aadA-.assembled</th>\n",
       "      <th>aadA1+.assembled</th>\n",
       "      <th>aadA2+.assembled</th>\n",
       "      <th>ant_2____Ia.assembled</th>\n",
       "      <th>aphA16.assembled</th>\n",
       "      <th>aph_3___Ia.assembled</th>\n",
       "      <th>aph_3___XV.assembled</th>\n",
       "      <th>...</th>\n",
       "      <th>qnrS-.assembled</th>\n",
       "      <th>sat2+.assembled</th>\n",
       "      <th>sul1.assembled</th>\n",
       "      <th>sul2.assembled</th>\n",
       "      <th>tet_C_.assembled</th>\n",
       "      <th>acrR.assembled</th>\n",
       "      <th>gyrA.assembled</th>\n",
       "      <th>ompK36.assembled</th>\n",
       "      <th>ompK37.assembled</th>\n",
       "      <th>ramR.assembled</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KPN12_S2_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>partial</td>\n",
       "      <td>partial</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN13_S3_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>partial</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>partial</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN14_S4_L001</th>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>partial</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN15_S5_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>interrupted</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>fragmented</td>\n",
       "      <td>yes</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNH_S7_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>partial</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNI_S8_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>partial</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>partial</td>\n",
       "      <td>no</td>\n",
       "      <td>partial</td>\n",
       "      <td>partial</td>\n",
       "      <td>no</td>\n",
       "      <td>partial</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>partial</td>\n",
       "      <td>partial</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>partial</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNJ_S2_L001</th>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>partial</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNK_S1_L001</th>\n",
       "      <td>partial</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>interrupted</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>partial</td>\n",
       "      <td>partial</td>\n",
       "      <td>yes</td>\n",
       "      <td>interrupted</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              A7J11_01233+.assembled aac_3__II-.assembled  \\\n",
       "samples                                                     \n",
       "KPN12_S2_L001                partial                   no   \n",
       "KPN13_S3_L001                partial                   no   \n",
       "KPN14_S4_L001                     no                   no   \n",
       "KPN15_S5_L001                partial        yes_nonunique   \n",
       "KPNH_S7_L001                 partial                   no   \n",
       "KPNI_S8_L001                 partial              partial   \n",
       "KPNJ_S2_L001                      no                   no   \n",
       "KPNK_S1_L001                 partial        yes_nonunique   \n",
       "\n",
       "              aac_6___Ib+.assembled aadA-.assembled aadA1+.assembled  \\\n",
       "samples                                                                \n",
       "KPN12_S2_L001                   yes              no               no   \n",
       "KPN13_S3_L001                   yes              no               no   \n",
       "KPN14_S4_L001                    no              no               no   \n",
       "KPN15_S5_L001         yes_nonunique     interrupted    yes_nonunique   \n",
       "KPNH_S7_L001                    yes              no               no   \n",
       "KPNI_S8_L001                    yes              no          partial   \n",
       "KPNJ_S2_L001                     no              no               no   \n",
       "KPNK_S1_L001          yes_nonunique     interrupted    yes_nonunique   \n",
       "\n",
       "              aadA2+.assembled ant_2____Ia.assembled aphA16.assembled  \\\n",
       "samples                                                                 \n",
       "KPN12_S2_L001    yes_nonunique                    no               no   \n",
       "KPN13_S3_L001               no                    no               no   \n",
       "KPN14_S4_L001               no                    no               no   \n",
       "KPN15_S5_L001               no                    no               no   \n",
       "KPNH_S7_L001     yes_nonunique                    no               no   \n",
       "KPNI_S8_L001                no               partial          partial   \n",
       "KPNJ_S2_L001                no                    no               no   \n",
       "KPNK_S1_L001                no                    no               no   \n",
       "\n",
       "              aph_3___Ia.assembled aph_3___XV.assembled  ... qnrS-.assembled  \\\n",
       "samples                                                  ...                   \n",
       "KPN12_S2_L001                  yes                   no  ...              no   \n",
       "KPN13_S3_L001                   no              partial  ...             yes   \n",
       "KPN14_S4_L001                  yes                   no  ...              no   \n",
       "KPN15_S5_L001                   no                   no  ...              no   \n",
       "KPNH_S7_L001                   yes                   no  ...              no   \n",
       "KPNI_S8_L001                    no              partial  ...             yes   \n",
       "KPNJ_S2_L001                   yes                   no  ...              no   \n",
       "KPNK_S1_L001                    no                   no  ...              no   \n",
       "\n",
       "              sat2+.assembled sul1.assembled sul2.assembled tet_C_.assembled  \\\n",
       "samples                                                                        \n",
       "KPN12_S2_L001              no            yes             no               no   \n",
       "KPN13_S3_L001              no             no             no               no   \n",
       "KPN14_S4_L001              no            yes             no              yes   \n",
       "KPN15_S5_L001             yes            yes            yes               no   \n",
       "KPNH_S7_L001               no            yes             no               no   \n",
       "KPNI_S8_L001               no        partial        partial               no   \n",
       "KPNJ_S2_L001               no            yes             no              yes   \n",
       "KPNK_S1_L001              yes            yes            yes               no   \n",
       "\n",
       "              acrR.assembled gyrA.assembled ompK36.assembled ompK37.assembled  \\\n",
       "samples                                                                         \n",
       "KPN12_S2_L001            yes        partial          partial              yes   \n",
       "KPN13_S3_L001            yes        partial              yes              yes   \n",
       "KPN14_S4_L001            yes        partial              yes              yes   \n",
       "KPN15_S5_L001            yes             no       fragmented              yes   \n",
       "KPNH_S7_L001             yes        partial              yes              yes   \n",
       "KPNI_S8_L001             yes        partial              yes              yes   \n",
       "KPNJ_S2_L001             yes        partial              yes              yes   \n",
       "KPNK_S1_L001             yes        partial          partial              yes   \n",
       "\n",
       "              ramR.assembled  \n",
       "samples                       \n",
       "KPN12_S2_L001            yes  \n",
       "KPN13_S3_L001    interrupted  \n",
       "KPN14_S4_L001    interrupted  \n",
       "KPN15_S5_L001    interrupted  \n",
       "KPNH_S7_L001             yes  \n",
       "KPNI_S8_L001     interrupted  \n",
       "KPNJ_S2_L001     interrupted  \n",
       "KPNK_S1_L001     interrupted  \n",
       "\n",
       "[8 rows x 54 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "asd=datos.iloc[:,AS]\n",
    "asd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A7J11_01233+.assembled</th>\n",
       "      <th>aac_3__II-.assembled</th>\n",
       "      <th>aac_6___Ib+.assembled</th>\n",
       "      <th>aadA-.assembled</th>\n",
       "      <th>aadA1+.assembled</th>\n",
       "      <th>aadA2+.assembled</th>\n",
       "      <th>ant_2____Ia.assembled</th>\n",
       "      <th>aphA16.assembled</th>\n",
       "      <th>aph_3___Ia.assembled</th>\n",
       "      <th>aph_3___XV.assembled</th>\n",
       "      <th>...</th>\n",
       "      <th>qnrS-.assembled</th>\n",
       "      <th>sat2+.assembled</th>\n",
       "      <th>sul1.assembled</th>\n",
       "      <th>sul2.assembled</th>\n",
       "      <th>tet_C_.assembled</th>\n",
       "      <th>acrR.assembled</th>\n",
       "      <th>gyrA.assembled</th>\n",
       "      <th>ompK36.assembled</th>\n",
       "      <th>ompK37.assembled</th>\n",
       "      <th>ramR.assembled</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KPN12_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN13_S3_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN14_S4_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN15_S5_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNH_S7_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNI_S8_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNJ_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNK_S1_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              A7J11_01233+.assembled aac_3__II-.assembled  \\\n",
       "samples                                                     \n",
       "KPN12_S2_L001                    NaN                  NaN   \n",
       "KPN13_S3_L001                    NaN                  NaN   \n",
       "KPN14_S4_L001                    NaN                  NaN   \n",
       "KPN15_S5_L001                    NaN                  NaN   \n",
       "KPNH_S7_L001                     NaN                  NaN   \n",
       "KPNI_S8_L001                     NaN                  NaN   \n",
       "KPNJ_S2_L001                     NaN                  NaN   \n",
       "KPNK_S1_L001                     NaN                  NaN   \n",
       "\n",
       "              aac_6___Ib+.assembled aadA-.assembled aadA1+.assembled  \\\n",
       "samples                                                                \n",
       "KPN12_S2_L001                   yes             NaN              NaN   \n",
       "KPN13_S3_L001                   yes             NaN              NaN   \n",
       "KPN14_S4_L001                   NaN             NaN              NaN   \n",
       "KPN15_S5_L001                   NaN             NaN              NaN   \n",
       "KPNH_S7_L001                    yes             NaN              NaN   \n",
       "KPNI_S8_L001                    yes             NaN              NaN   \n",
       "KPNJ_S2_L001                    NaN             NaN              NaN   \n",
       "KPNK_S1_L001                    NaN             NaN              NaN   \n",
       "\n",
       "              aadA2+.assembled ant_2____Ia.assembled aphA16.assembled  \\\n",
       "samples                                                                 \n",
       "KPN12_S2_L001              NaN                   NaN              NaN   \n",
       "KPN13_S3_L001              NaN                   NaN              NaN   \n",
       "KPN14_S4_L001              NaN                   NaN              NaN   \n",
       "KPN15_S5_L001              NaN                   NaN              NaN   \n",
       "KPNH_S7_L001               NaN                   NaN              NaN   \n",
       "KPNI_S8_L001               NaN                   NaN              NaN   \n",
       "KPNJ_S2_L001               NaN                   NaN              NaN   \n",
       "KPNK_S1_L001               NaN                   NaN              NaN   \n",
       "\n",
       "              aph_3___Ia.assembled aph_3___XV.assembled  ... qnrS-.assembled  \\\n",
       "samples                                                  ...                   \n",
       "KPN12_S2_L001                  yes                  NaN  ...             NaN   \n",
       "KPN13_S3_L001                  NaN                  NaN  ...             yes   \n",
       "KPN14_S4_L001                  yes                  NaN  ...             NaN   \n",
       "KPN15_S5_L001                  NaN                  NaN  ...             NaN   \n",
       "KPNH_S7_L001                   yes                  NaN  ...             NaN   \n",
       "KPNI_S8_L001                   NaN                  NaN  ...             yes   \n",
       "KPNJ_S2_L001                   yes                  NaN  ...             NaN   \n",
       "KPNK_S1_L001                   NaN                  NaN  ...             NaN   \n",
       "\n",
       "              sat2+.assembled sul1.assembled sul2.assembled tet_C_.assembled  \\\n",
       "samples                                                                        \n",
       "KPN12_S2_L001             NaN            yes            NaN              NaN   \n",
       "KPN13_S3_L001             NaN            NaN            NaN              NaN   \n",
       "KPN14_S4_L001             NaN            yes            NaN              yes   \n",
       "KPN15_S5_L001             yes            yes            yes              NaN   \n",
       "KPNH_S7_L001              NaN            yes            NaN              NaN   \n",
       "KPNI_S8_L001              NaN            NaN            NaN              NaN   \n",
       "KPNJ_S2_L001              NaN            yes            NaN              yes   \n",
       "KPNK_S1_L001              yes            yes            yes              NaN   \n",
       "\n",
       "              acrR.assembled gyrA.assembled ompK36.assembled ompK37.assembled  \\\n",
       "samples                                                                         \n",
       "KPN12_S2_L001            yes            NaN              NaN              yes   \n",
       "KPN13_S3_L001            yes            NaN              yes              yes   \n",
       "KPN14_S4_L001            yes            NaN              yes              yes   \n",
       "KPN15_S5_L001            yes            NaN              NaN              yes   \n",
       "KPNH_S7_L001             yes            NaN              yes              yes   \n",
       "KPNI_S8_L001             yes            NaN              yes              yes   \n",
       "KPNJ_S2_L001             yes            NaN              yes              yes   \n",
       "KPNK_S1_L001             yes            NaN              NaN              yes   \n",
       "\n",
       "              ramR.assembled  \n",
       "samples                       \n",
       "KPN12_S2_L001            yes  \n",
       "KPN13_S3_L001            NaN  \n",
       "KPN14_S4_L001            NaN  \n",
       "KPN15_S5_L001            NaN  \n",
       "KPNH_S7_L001             yes  \n",
       "KPNI_S8_L001             NaN  \n",
       "KPNJ_S2_L001             NaN  \n",
       "KPNK_S1_L001             NaN  \n",
       "\n",
       "[8 rows x 54 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "yes=asd[asd==\"yes\"]\n",
    "yes\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A7J11_01233+.assembled</th>\n",
       "      <th>aac_3__II-.assembled</th>\n",
       "      <th>aac_6___Ib+.assembled</th>\n",
       "      <th>aadA-.assembled</th>\n",
       "      <th>aadA1+.assembled</th>\n",
       "      <th>aadA2+.assembled</th>\n",
       "      <th>ant_2____Ia.assembled</th>\n",
       "      <th>aphA16.assembled</th>\n",
       "      <th>aph_3___Ia.assembled</th>\n",
       "      <th>aph_3___XV.assembled</th>\n",
       "      <th>...</th>\n",
       "      <th>qnrS-.assembled</th>\n",
       "      <th>sat2+.assembled</th>\n",
       "      <th>sul1.assembled</th>\n",
       "      <th>sul2.assembled</th>\n",
       "      <th>tet_C_.assembled</th>\n",
       "      <th>acrR.assembled</th>\n",
       "      <th>gyrA.assembled</th>\n",
       "      <th>ompK36.assembled</th>\n",
       "      <th>ompK37.assembled</th>\n",
       "      <th>ramR.assembled</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KPN12_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN13_S3_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN14_S4_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN15_S5_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNH_S7_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNI_S8_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNJ_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNK_S1_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              A7J11_01233+.assembled aac_3__II-.assembled  \\\n",
       "samples                                                     \n",
       "KPN12_S2_L001                    NaN                  NaN   \n",
       "KPN13_S3_L001                    NaN                  NaN   \n",
       "KPN14_S4_L001                    NaN                  NaN   \n",
       "KPN15_S5_L001                    NaN        yes_nonunique   \n",
       "KPNH_S7_L001                     NaN                  NaN   \n",
       "KPNI_S8_L001                     NaN                  NaN   \n",
       "KPNJ_S2_L001                     NaN                  NaN   \n",
       "KPNK_S1_L001                     NaN        yes_nonunique   \n",
       "\n",
       "              aac_6___Ib+.assembled aadA-.assembled aadA1+.assembled  \\\n",
       "samples                                                                \n",
       "KPN12_S2_L001                   NaN             NaN              NaN   \n",
       "KPN13_S3_L001                   NaN             NaN              NaN   \n",
       "KPN14_S4_L001                   NaN             NaN              NaN   \n",
       "KPN15_S5_L001         yes_nonunique             NaN    yes_nonunique   \n",
       "KPNH_S7_L001                    NaN             NaN              NaN   \n",
       "KPNI_S8_L001                    NaN             NaN              NaN   \n",
       "KPNJ_S2_L001                    NaN             NaN              NaN   \n",
       "KPNK_S1_L001          yes_nonunique             NaN    yes_nonunique   \n",
       "\n",
       "              aadA2+.assembled ant_2____Ia.assembled aphA16.assembled  \\\n",
       "samples                                                                 \n",
       "KPN12_S2_L001    yes_nonunique                   NaN              NaN   \n",
       "KPN13_S3_L001              NaN                   NaN              NaN   \n",
       "KPN14_S4_L001              NaN                   NaN              NaN   \n",
       "KPN15_S5_L001              NaN                   NaN              NaN   \n",
       "KPNH_S7_L001     yes_nonunique                   NaN              NaN   \n",
       "KPNI_S8_L001               NaN                   NaN              NaN   \n",
       "KPNJ_S2_L001               NaN                   NaN              NaN   \n",
       "KPNK_S1_L001               NaN                   NaN              NaN   \n",
       "\n",
       "              aph_3___Ia.assembled aph_3___XV.assembled  ... qnrS-.assembled  \\\n",
       "samples                                                  ...                   \n",
       "KPN12_S2_L001                  NaN                  NaN  ...             NaN   \n",
       "KPN13_S3_L001                  NaN                  NaN  ...             NaN   \n",
       "KPN14_S4_L001                  NaN                  NaN  ...             NaN   \n",
       "KPN15_S5_L001                  NaN                  NaN  ...             NaN   \n",
       "KPNH_S7_L001                   NaN                  NaN  ...             NaN   \n",
       "KPNI_S8_L001                   NaN                  NaN  ...             NaN   \n",
       "KPNJ_S2_L001                   NaN                  NaN  ...             NaN   \n",
       "KPNK_S1_L001                   NaN                  NaN  ...             NaN   \n",
       "\n",
       "              sat2+.assembled sul1.assembled sul2.assembled tet_C_.assembled  \\\n",
       "samples                                                                        \n",
       "KPN12_S2_L001             NaN            NaN            NaN              NaN   \n",
       "KPN13_S3_L001             NaN            NaN            NaN              NaN   \n",
       "KPN14_S4_L001             NaN            NaN            NaN              NaN   \n",
       "KPN15_S5_L001             NaN            NaN            NaN              NaN   \n",
       "KPNH_S7_L001              NaN            NaN            NaN              NaN   \n",
       "KPNI_S8_L001              NaN            NaN            NaN              NaN   \n",
       "KPNJ_S2_L001              NaN            NaN            NaN              NaN   \n",
       "KPNK_S1_L001              NaN            NaN            NaN              NaN   \n",
       "\n",
       "              acrR.assembled gyrA.assembled ompK36.assembled ompK37.assembled  \\\n",
       "samples                                                                         \n",
       "KPN12_S2_L001            NaN            NaN              NaN              NaN   \n",
       "KPN13_S3_L001            NaN            NaN              NaN              NaN   \n",
       "KPN14_S4_L001            NaN            NaN              NaN              NaN   \n",
       "KPN15_S5_L001            NaN            NaN              NaN              NaN   \n",
       "KPNH_S7_L001             NaN            NaN              NaN              NaN   \n",
       "KPNI_S8_L001             NaN            NaN              NaN              NaN   \n",
       "KPNJ_S2_L001             NaN            NaN              NaN              NaN   \n",
       "KPNK_S1_L001             NaN            NaN              NaN              NaN   \n",
       "\n",
       "              ramR.assembled  \n",
       "samples                       \n",
       "KPN12_S2_L001            NaN  \n",
       "KPN13_S3_L001            NaN  \n",
       "KPN14_S4_L001            NaN  \n",
       "KPN15_S5_L001            NaN  \n",
       "KPNH_S7_L001             NaN  \n",
       "KPNI_S8_L001             NaN  \n",
       "KPNJ_S2_L001             NaN  \n",
       "KPNK_S1_L001             NaN  \n",
       "\n",
       "[8 rows x 54 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "yes_nonunique=asd[asd==\"yes_nonunique\"]\n",
    "yes_nonunique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A7J11_01233+.assembled</th>\n",
       "      <th>aac_3__II-.assembled</th>\n",
       "      <th>aac_6___Ib+.assembled</th>\n",
       "      <th>aadA-.assembled</th>\n",
       "      <th>aadA1+.assembled</th>\n",
       "      <th>aadA2+.assembled</th>\n",
       "      <th>ant_2____Ia.assembled</th>\n",
       "      <th>aphA16.assembled</th>\n",
       "      <th>aph_3___Ia.assembled</th>\n",
       "      <th>aph_3___XV.assembled</th>\n",
       "      <th>...</th>\n",
       "      <th>qnrS-.assembled</th>\n",
       "      <th>sat2+.assembled</th>\n",
       "      <th>sul1.assembled</th>\n",
       "      <th>sul2.assembled</th>\n",
       "      <th>tet_C_.assembled</th>\n",
       "      <th>acrR.assembled</th>\n",
       "      <th>gyrA.assembled</th>\n",
       "      <th>ompK36.assembled</th>\n",
       "      <th>ompK37.assembled</th>\n",
       "      <th>ramR.assembled</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KPN12_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN13_S3_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN14_S4_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN15_S5_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNH_S7_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNI_S8_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNJ_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNK_S1_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 108 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              A7J11_01233+.assembled aac_3__II-.assembled  \\\n",
       "samples                                                     \n",
       "KPN12_S2_L001                    NaN                  NaN   \n",
       "KPN13_S3_L001                    NaN                  NaN   \n",
       "KPN14_S4_L001                    NaN                  NaN   \n",
       "KPN15_S5_L001                    NaN                  NaN   \n",
       "KPNH_S7_L001                     NaN                  NaN   \n",
       "KPNI_S8_L001                     NaN                  NaN   \n",
       "KPNJ_S2_L001                     NaN                  NaN   \n",
       "KPNK_S1_L001                     NaN                  NaN   \n",
       "\n",
       "              aac_6___Ib+.assembled aadA-.assembled aadA1+.assembled  \\\n",
       "samples                                                                \n",
       "KPN12_S2_L001                   yes             NaN              NaN   \n",
       "KPN13_S3_L001                   yes             NaN              NaN   \n",
       "KPN14_S4_L001                   NaN             NaN              NaN   \n",
       "KPN15_S5_L001                   NaN             NaN              NaN   \n",
       "KPNH_S7_L001                    yes             NaN              NaN   \n",
       "KPNI_S8_L001                    yes             NaN              NaN   \n",
       "KPNJ_S2_L001                    NaN             NaN              NaN   \n",
       "KPNK_S1_L001                    NaN             NaN              NaN   \n",
       "\n",
       "              aadA2+.assembled ant_2____Ia.assembled aphA16.assembled  \\\n",
       "samples                                                                 \n",
       "KPN12_S2_L001              NaN                   NaN              NaN   \n",
       "KPN13_S3_L001              NaN                   NaN              NaN   \n",
       "KPN14_S4_L001              NaN                   NaN              NaN   \n",
       "KPN15_S5_L001              NaN                   NaN              NaN   \n",
       "KPNH_S7_L001               NaN                   NaN              NaN   \n",
       "KPNI_S8_L001               NaN                   NaN              NaN   \n",
       "KPNJ_S2_L001               NaN                   NaN              NaN   \n",
       "KPNK_S1_L001               NaN                   NaN              NaN   \n",
       "\n",
       "              aph_3___Ia.assembled aph_3___XV.assembled  ... qnrS-.assembled  \\\n",
       "samples                                                  ...                   \n",
       "KPN12_S2_L001                  yes                  NaN  ...             NaN   \n",
       "KPN13_S3_L001                  NaN                  NaN  ...             NaN   \n",
       "KPN14_S4_L001                  yes                  NaN  ...             NaN   \n",
       "KPN15_S5_L001                  NaN                  NaN  ...             NaN   \n",
       "KPNH_S7_L001                   yes                  NaN  ...             NaN   \n",
       "KPNI_S8_L001                   NaN                  NaN  ...             NaN   \n",
       "KPNJ_S2_L001                   yes                  NaN  ...             NaN   \n",
       "KPNK_S1_L001                   NaN                  NaN  ...             NaN   \n",
       "\n",
       "              sat2+.assembled sul1.assembled sul2.assembled tet_C_.assembled  \\\n",
       "samples                                                                        \n",
       "KPN12_S2_L001             NaN            NaN            NaN              NaN   \n",
       "KPN13_S3_L001             NaN            NaN            NaN              NaN   \n",
       "KPN14_S4_L001             NaN            NaN            NaN              NaN   \n",
       "KPN15_S5_L001             NaN            NaN            NaN              NaN   \n",
       "KPNH_S7_L001              NaN            NaN            NaN              NaN   \n",
       "KPNI_S8_L001              NaN            NaN            NaN              NaN   \n",
       "KPNJ_S2_L001              NaN            NaN            NaN              NaN   \n",
       "KPNK_S1_L001              NaN            NaN            NaN              NaN   \n",
       "\n",
       "              acrR.assembled gyrA.assembled ompK36.assembled ompK37.assembled  \\\n",
       "samples                                                                         \n",
       "KPN12_S2_L001            NaN            NaN              NaN              NaN   \n",
       "KPN13_S3_L001            NaN            NaN              NaN              NaN   \n",
       "KPN14_S4_L001            NaN            NaN              NaN              NaN   \n",
       "KPN15_S5_L001            NaN            NaN              NaN              NaN   \n",
       "KPNH_S7_L001             NaN            NaN              NaN              NaN   \n",
       "KPNI_S8_L001             NaN            NaN              NaN              NaN   \n",
       "KPNJ_S2_L001             NaN            NaN              NaN              NaN   \n",
       "KPNK_S1_L001             NaN            NaN              NaN              NaN   \n",
       "\n",
       "              ramR.assembled  \n",
       "samples                       \n",
       "KPN12_S2_L001            NaN  \n",
       "KPN13_S3_L001            NaN  \n",
       "KPN14_S4_L001            NaN  \n",
       "KPN15_S5_L001            NaN  \n",
       "KPNH_S7_L001             NaN  \n",
       "KPNI_S8_L001             NaN  \n",
       "KPNJ_S2_L001             NaN  \n",
       "KPNK_S1_L001             NaN  \n",
       "\n",
       "[8 rows x 108 columns]"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "yeah= pd.concat([yes, yes_nonunique], axis=1)\n",
    "yeah"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "samples                             \n",
       "KPN12_S2_L001  aac_6___Ib+.assembled              yes\n",
       "               aph_3___Ia.assembled               yes\n",
       "               bla-_6.assembled                   yes\n",
       "               blaKPC_-.assembled                 yes\n",
       "               blaTEM_-.assembled                 yes\n",
       "                                            ...      \n",
       "KPNK_S1_L001   aac_3__II-.assembled     yes_nonunique\n",
       "               aac_6___Ib+.assembled    yes_nonunique\n",
       "               aadA1+.assembled         yes_nonunique\n",
       "               bla-_12.assembled        yes_nonunique\n",
       "               blaTEM_-.assembled       yes_nonunique\n",
       "Length: 151, dtype: object"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "yeah_column=yeah.stack()\n",
    "yeah_column    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MultiIndex([('KPN12_S2_L001', 'aac_6___Ib+.assembled'),\n",
       "            ('KPN12_S2_L001',  'aph_3___Ia.assembled'),\n",
       "            ('KPN12_S2_L001',      'bla-_6.assembled'),\n",
       "            ('KPN12_S2_L001',    'blaKPC_-.assembled'),\n",
       "            ('KPN12_S2_L001',    'blaTEM_-.assembled'),\n",
       "            ('KPN12_S2_L001',       'catA1.assembled'),\n",
       "            ('KPN12_S2_L001',      'dfrA12.assembled'),\n",
       "            ('KPN12_S2_L001',     'fosA-_1.assembled'),\n",
       "            ('KPN12_S2_L001',      'mph_A_.assembled'),\n",
       "            ('KPN12_S2_L001',       'oqxA-.assembled'),\n",
       "            ...\n",
       "            ( 'KPNK_S1_L001',       'sat2+.assembled'),\n",
       "            ( 'KPNK_S1_L001',        'sul1.assembled'),\n",
       "            ( 'KPNK_S1_L001',        'sul2.assembled'),\n",
       "            ( 'KPNK_S1_L001',        'acrR.assembled'),\n",
       "            ( 'KPNK_S1_L001',      'ompK37.assembled'),\n",
       "            ( 'KPNK_S1_L001',  'aac_3__II-.assembled'),\n",
       "            ( 'KPNK_S1_L001', 'aac_6___Ib+.assembled'),\n",
       "            ( 'KPNK_S1_L001',      'aadA1+.assembled'),\n",
       "            ( 'KPNK_S1_L001',     'bla-_12.assembled'),\n",
       "            ( 'KPNK_S1_L001',    'blaTEM_-.assembled')],\n",
       "           names=['samples', None], length=151)"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "yeah_column.index\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aac_6___Ib+.assembled</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aph_3___Ia.assembled</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>bla-_6.assembled</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>blaKPC_-.assembled</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>blaTEM_-.assembled</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>aac_3__II-.assembled</td>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>aac_6___Ib+.assembled</td>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>aadA1+.assembled</td>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>bla-_12.assembled</td>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>blaTEM_-.assembled</td>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           samples                level_1              0\n",
       "0    KPN12_S2_L001  aac_6___Ib+.assembled            yes\n",
       "1    KPN12_S2_L001   aph_3___Ia.assembled            yes\n",
       "2    KPN12_S2_L001       bla-_6.assembled            yes\n",
       "3    KPN12_S2_L001     blaKPC_-.assembled            yes\n",
       "4    KPN12_S2_L001     blaTEM_-.assembled            yes\n",
       "..             ...                    ...            ...\n",
       "146   KPNK_S1_L001   aac_3__II-.assembled  yes_nonunique\n",
       "147   KPNK_S1_L001  aac_6___Ib+.assembled  yes_nonunique\n",
       "148   KPNK_S1_L001       aadA1+.assembled  yes_nonunique\n",
       "149   KPNK_S1_L001      bla-_12.assembled  yes_nonunique\n",
       "150   KPNK_S1_L001     blaTEM_-.assembled  yes_nonunique\n",
       "\n",
       "[151 rows x 3 columns]"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "yeah_column_df = yeah_column.reset_index(level=[0,1])\n",
    "yeah_column_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>level_1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>aac_6___Ib+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>aph_3___I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>bla-_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>blaKPC_-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>blaTEM_-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>aac_3__II-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>aac_6___Ib+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>aadA1+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>bla-_12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>blaTEM_-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         level_1\n",
       "0    aac_6___Ib+\n",
       "1      aph_3___I\n",
       "2         bla-_6\n",
       "3       blaKPC_-\n",
       "4       blaTEM_-\n",
       "..           ...\n",
       "146   aac_3__II-\n",
       "147  aac_6___Ib+\n",
       "148       aadA1+\n",
       "149      bla-_12\n",
       "150     blaTEM_-\n",
       "\n",
       "[151 rows x 1 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fixed_level1=yeah_column_df.level_1.str.rstrip('.assembled')\n",
    "fixed_level2=pd.DataFrame(fixed_level1)\n",
    "fixed_level2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aac_6___Ib+</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aph_3___I</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>bla-_6</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>blaKPC_-</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>blaTEM_-</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>aac_3__II-</td>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>aac_6___Ib+</td>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>aadA1+</td>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>bla-_12</td>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>blaTEM_-</td>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           samples      level_1              0\n",
       "0    KPN12_S2_L001  aac_6___Ib+            yes\n",
       "1    KPN12_S2_L001    aph_3___I            yes\n",
       "2    KPN12_S2_L001       bla-_6            yes\n",
       "3    KPN12_S2_L001     blaKPC_-            yes\n",
       "4    KPN12_S2_L001     blaTEM_-            yes\n",
       "..             ...          ...            ...\n",
       "146   KPNK_S1_L001   aac_3__II-  yes_nonunique\n",
       "147   KPNK_S1_L001  aac_6___Ib+  yes_nonunique\n",
       "148   KPNK_S1_L001       aadA1+  yes_nonunique\n",
       "149   KPNK_S1_L001      bla-_12  yes_nonunique\n",
       "150   KPNK_S1_L001     blaTEM_-  yes_nonunique\n",
       "\n",
       "[151 rows x 3 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "yeah_column_df['level_1']=fixed_level2['level_1']\n",
    "yeah_column_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>yes/yes_nonunique</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPN12_S2_L001</th>\n",
       "      <th>aac_6___Ib+</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aph_3___I</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bla-_6</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>blaKPC_-</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>blaTEM_-</th>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPNK_S1_L001</th>\n",
       "      <th>aac_3__II-</th>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aac_6___Ib+</th>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aadA1+</th>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bla-_12</th>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>blaTEM_-</th>\n",
       "      <td>yes_nonunique</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          yes/yes_nonunique\n",
       "samples       level_1                      \n",
       "KPN12_S2_L001 aac_6___Ib+               yes\n",
       "              aph_3___I                 yes\n",
       "              bla-_6                    yes\n",
       "              blaKPC_-                  yes\n",
       "              blaTEM_-                  yes\n",
       "...                                     ...\n",
       "KPNK_S1_L001  aac_3__II-      yes_nonunique\n",
       "              aac_6___Ib+     yes_nonunique\n",
       "              aadA1+          yes_nonunique\n",
       "              bla-_12         yes_nonunique\n",
       "              blaTEM_-        yes_nonunique\n",
       "\n",
       "[151 rows x 1 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#####----------------------------------> data frame 1 ---> assembled arreglados!!\n",
    "yeah_column_arreglada=yeah_column_df.set_index(['samples', 'level_1'])\n",
    "yeah_column_arreglada=yeah_column_arreglada.rename(columns = {0:'yes/yes_nonunique'}, inplace = False)\n",
    "yeah_column_arreglada"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Int64Index([-1, -1, 12, -1, -1, -1, 10, -1, -1, -1,\n",
       "            ...\n",
       "            -1, -1, -1, -1, -1, -1, -1, -1, -1, -1],\n",
       "           dtype='int64', length=232)"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##### agroupment of the \".ctq_cov\" data \n",
    "gene_id=datos.columns.str.find(\".pct_id\")\n",
    "gene_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "       False, False, False, False, False, False, False, False, False,\n",
       "       False, False, False, False, False, False, False, False, False,\n",
       "       False, False, False, False, False, False, False, False, False,\n",
       "       False, False, False, False, False, False, False])"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ID=gene_id != -1\n",
    "ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A7J11_01233+.pct_id</th>\n",
       "      <th>aac_3__II-.pct_id</th>\n",
       "      <th>aac_6___Ib+.pct_id</th>\n",
       "      <th>aadA-.pct_id</th>\n",
       "      <th>aadA1+.pct_id</th>\n",
       "      <th>aadA2+.pct_id</th>\n",
       "      <th>ant_2____Ia.pct_id</th>\n",
       "      <th>aphA16.pct_id</th>\n",
       "      <th>aph_3___Ia.pct_id</th>\n",
       "      <th>aph_3___XV.pct_id</th>\n",
       "      <th>...</th>\n",
       "      <th>mph_E_.pct_id</th>\n",
       "      <th>msr_E_.pct_id</th>\n",
       "      <th>oqxA-.pct_id</th>\n",
       "      <th>oqxB-.pct_id</th>\n",
       "      <th>qacE-.pct_id</th>\n",
       "      <th>qnrS-.pct_id</th>\n",
       "      <th>sat2+.pct_id</th>\n",
       "      <th>sul1.pct_id</th>\n",
       "      <th>sul2.pct_id</th>\n",
       "      <th>tet_C_.pct_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KPN12_S2_L001</th>\n",
       "      <td>99.83</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.87</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN13_S3_L001</th>\n",
       "      <td>99.65</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.82</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.57</td>\n",
       "      <td>99.71</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN14_S4_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.66</td>\n",
       "      <td>99.43</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN15_S5_L001</th>\n",
       "      <td>99.83</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.00</td>\n",
       "      <td>91.74</td>\n",
       "      <td>99.87</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>100.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>99.74</td>\n",
       "      <td>99.90</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNH_S7_L001</th>\n",
       "      <td>99.83</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNI_S8_L001</th>\n",
       "      <td>99.65</td>\n",
       "      <td>99.76</td>\n",
       "      <td>99.82</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>99.61</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.57</td>\n",
       "      <td>99.71</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.75</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNJ_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.66</td>\n",
       "      <td>99.43</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNK_S1_L001</th>\n",
       "      <td>99.83</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.00</td>\n",
       "      <td>91.86</td>\n",
       "      <td>99.87</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>100.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>99.74</td>\n",
       "      <td>99.90</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               A7J11_01233+.pct_id  aac_3__II-.pct_id  aac_6___Ib+.pct_id  \\\n",
       "samples                                                                     \n",
       "KPN12_S2_L001                99.83                NaN              100.00   \n",
       "KPN13_S3_L001                99.65                NaN               99.82   \n",
       "KPN14_S4_L001                  NaN                NaN                 NaN   \n",
       "KPN15_S5_L001                99.83             100.00              100.00   \n",
       "KPNH_S7_L001                 99.83                NaN              100.00   \n",
       "KPNI_S8_L001                 99.65              99.76               99.82   \n",
       "KPNJ_S2_L001                   NaN                NaN                 NaN   \n",
       "KPNK_S1_L001                 99.83             100.00              100.00   \n",
       "\n",
       "               aadA-.pct_id  aadA1+.pct_id  aadA2+.pct_id  ant_2____Ia.pct_id  \\\n",
       "samples                                                                         \n",
       "KPN12_S2_L001           NaN            NaN          99.87                 NaN   \n",
       "KPN13_S3_L001           NaN            NaN            NaN                 NaN   \n",
       "KPN14_S4_L001           NaN            NaN            NaN                 NaN   \n",
       "KPN15_S5_L001         91.74          99.87            NaN                 NaN   \n",
       "KPNH_S7_L001            NaN            NaN         100.00                 NaN   \n",
       "KPNI_S8_L001            NaN         100.00            NaN               100.0   \n",
       "KPNJ_S2_L001            NaN            NaN            NaN                 NaN   \n",
       "KPNK_S1_L001          91.86          99.87            NaN                 NaN   \n",
       "\n",
       "               aphA16.pct_id  aph_3___Ia.pct_id  aph_3___XV.pct_id  ...  \\\n",
       "samples                                                             ...   \n",
       "KPN12_S2_L001            NaN              100.0                NaN  ...   \n",
       "KPN13_S3_L001            NaN                NaN              100.0  ...   \n",
       "KPN14_S4_L001            NaN              100.0                NaN  ...   \n",
       "KPN15_S5_L001            NaN                NaN                NaN  ...   \n",
       "KPNH_S7_L001             NaN              100.0                NaN  ...   \n",
       "KPNI_S8_L001           99.61                NaN              100.0  ...   \n",
       "KPNJ_S2_L001             NaN              100.0                NaN  ...   \n",
       "KPNK_S1_L001             NaN                NaN                NaN  ...   \n",
       "\n",
       "               mph_E_.pct_id  msr_E_.pct_id  oqxA-.pct_id  oqxB-.pct_id  \\\n",
       "samples                                                                   \n",
       "KPN12_S2_L001            NaN            NaN        100.00        100.00   \n",
       "KPN13_S3_L001            NaN            NaN         99.57         99.71   \n",
       "KPN14_S4_L001            NaN            NaN         99.66         99.43   \n",
       "KPN15_S5_L001          100.0          100.0         99.74         99.90   \n",
       "KPNH_S7_L001             NaN            NaN        100.00        100.00   \n",
       "KPNI_S8_L001             NaN            NaN         99.57         99.71   \n",
       "KPNJ_S2_L001             NaN            NaN         99.66         99.43   \n",
       "KPNK_S1_L001           100.0          100.0         99.74         99.90   \n",
       "\n",
       "               qacE-.pct_id  qnrS-.pct_id  sat2+.pct_id  sul1.pct_id  \\\n",
       "samples                                                                \n",
       "KPN12_S2_L001         100.0           NaN           NaN       100.00   \n",
       "KPN13_S3_L001           NaN         100.0           NaN          NaN   \n",
       "KPN14_S4_L001         100.0           NaN           NaN       100.00   \n",
       "KPN15_S5_L001         100.0           NaN         100.0       100.00   \n",
       "KPNH_S7_L001          100.0           NaN           NaN       100.00   \n",
       "KPNI_S8_L001            NaN         100.0           NaN        99.75   \n",
       "KPNJ_S2_L001          100.0           NaN           NaN       100.00   \n",
       "KPNK_S1_L001          100.0           NaN         100.0       100.00   \n",
       "\n",
       "               sul2.pct_id  tet_C_.pct_id  \n",
       "samples                                    \n",
       "KPN12_S2_L001          NaN            NaN  \n",
       "KPN13_S3_L001          NaN            NaN  \n",
       "KPN14_S4_L001          NaN          100.0  \n",
       "KPN15_S5_L001        100.0            NaN  \n",
       "KPNH_S7_L001           NaN            NaN  \n",
       "KPNI_S8_L001         100.0            NaN  \n",
       "KPNJ_S2_L001           NaN          100.0  \n",
       "KPNK_S1_L001         100.0            NaN  \n",
       "\n",
       "[8 rows x 49 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_identification=datos.iloc[:,ID]\n",
    "gene_identification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "samples                           \n",
       "KPN12_S2_L001  A7J11_01233+.pct_id     99.83\n",
       "               aac_6___Ib+.pct_id     100.00\n",
       "               aadA2+.pct_id           99.87\n",
       "               aph_3___Ia.pct_id      100.00\n",
       "               bla-_6.pct_id           99.88\n",
       "                                       ...  \n",
       "KPNK_S1_L001   oqxB-.pct_id            99.90\n",
       "               qacE-.pct_id           100.00\n",
       "               sat2+.pct_id           100.00\n",
       "               sul1.pct_id            100.00\n",
       "               sul2.pct_id            100.00\n",
       "Length: 155, dtype: float64"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_identification_df=gene_identification.stack()\n",
    "gene_identification_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>A7J11_01233+.pct_id</td>\n",
       "      <td>99.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aac_6___Ib+.pct_id</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aadA2+.pct_id</td>\n",
       "      <td>99.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aph_3___Ia.pct_id</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>bla-_6.pct_id</td>\n",
       "      <td>99.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>oqxB-.pct_id</td>\n",
       "      <td>99.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>qacE-.pct_id</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sat2+.pct_id</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sul1.pct_id</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sul2.pct_id</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>155 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           samples              level_1       0\n",
       "0    KPN12_S2_L001  A7J11_01233+.pct_id   99.83\n",
       "1    KPN12_S2_L001   aac_6___Ib+.pct_id  100.00\n",
       "2    KPN12_S2_L001        aadA2+.pct_id   99.87\n",
       "3    KPN12_S2_L001    aph_3___Ia.pct_id  100.00\n",
       "4    KPN12_S2_L001        bla-_6.pct_id   99.88\n",
       "..             ...                  ...     ...\n",
       "150   KPNK_S1_L001         oqxB-.pct_id   99.90\n",
       "151   KPNK_S1_L001         qacE-.pct_id  100.00\n",
       "152   KPNK_S1_L001         sat2+.pct_id  100.00\n",
       "153   KPNK_S1_L001          sul1.pct_id  100.00\n",
       "154   KPNK_S1_L001          sul2.pct_id  100.00\n",
       "\n",
       "[155 rows x 3 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_identification_df = gene_identification_df.reset_index(level=[0,1])\n",
    "gene_identification_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>level_1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A7J11_01233+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>aac_6___Ib+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>aadA2+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>aph_3___Ia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bla-_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>oqxB-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>qacE-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>sat2+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>sul1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>sul2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>155 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          level_1\n",
       "0    A7J11_01233+\n",
       "1     aac_6___Ib+\n",
       "2          aadA2+\n",
       "3      aph_3___Ia\n",
       "4          bla-_6\n",
       "..            ...\n",
       "150         oqxB-\n",
       "151         qacE-\n",
       "152         sat2+\n",
       "153          sul1\n",
       "154          sul2\n",
       "\n",
       "[155 rows x 1 columns]"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fixed_ID_level1_df=gene_identification_df.level_1.str.rstrip('.pct_id')\n",
    "fixed_ID_level1_df=pd.DataFrame(fixed_ID_level1_df)\n",
    "fixed_ID_level1_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>A7J11_01233+</td>\n",
       "      <td>99.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aac_6___Ib+</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aadA2+</td>\n",
       "      <td>99.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aph_3___Ia</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>bla-_6</td>\n",
       "      <td>99.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>oqxB-</td>\n",
       "      <td>99.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>qacE-</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sat2+</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sul1</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sul2</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>155 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           samples       level_1       0\n",
       "0    KPN12_S2_L001  A7J11_01233+   99.83\n",
       "1    KPN12_S2_L001   aac_6___Ib+  100.00\n",
       "2    KPN12_S2_L001        aadA2+   99.87\n",
       "3    KPN12_S2_L001    aph_3___Ia  100.00\n",
       "4    KPN12_S2_L001        bla-_6   99.88\n",
       "..             ...           ...     ...\n",
       "150   KPNK_S1_L001         oqxB-   99.90\n",
       "151   KPNK_S1_L001         qacE-  100.00\n",
       "152   KPNK_S1_L001         sat2+  100.00\n",
       "153   KPNK_S1_L001          sul1  100.00\n",
       "154   KPNK_S1_L001          sul2  100.00\n",
       "\n",
       "[155 rows x 3 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_identification_df['level_1']=fixed_ID_level1_df['level_1']\n",
    "gene_identification_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>%_ID</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPN12_S2_L001</th>\n",
       "      <th>A7J11_01233+</th>\n",
       "      <td>99.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aac_6___Ib+</th>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aadA2+</th>\n",
       "      <td>99.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aph_3___Ia</th>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bla-_6</th>\n",
       "      <td>99.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPNK_S1_L001</th>\n",
       "      <th>oqxB-</th>\n",
       "      <td>99.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>qacE-</th>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sat2+</th>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul1</th>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul2</th>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>155 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              %_ID\n",
       "samples       level_1             \n",
       "KPN12_S2_L001 A7J11_01233+   99.83\n",
       "              aac_6___Ib+   100.00\n",
       "              aadA2+         99.87\n",
       "              aph_3___Ia    100.00\n",
       "              bla-_6         99.88\n",
       "...                            ...\n",
       "KPNK_S1_L001  oqxB-          99.90\n",
       "              qacE-         100.00\n",
       "              sat2+         100.00\n",
       "              sul1          100.00\n",
       "              sul2          100.00\n",
       "\n",
       "[155 rows x 1 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#####----------------------------------> data frame 2 ---> identificacion de genes arreglados!!\n",
    "gene_identification_fixed=gene_identification_df.set_index(['samples', 'level_1'])\n",
    "gene_identification_fixed=gene_identification_fixed.rename(columns = {0:'%_ID'}, inplace = False)\n",
    "gene_identification_fixed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Int64Index([-1, -1, -1, 12, -1, -1, -1, 10, -1, -1,\n",
       "            ...\n",
       "            -1, -1, -1, -1, -1, -1, -1, -1, -1, -1],\n",
       "           dtype='int64', length=232)"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##### ------------------------------------------------>  agroupment of the \".ctq_cov\" data \n",
    "coverage=datos.columns.str.find(\".ctg_cov\")\n",
    "coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False, False, False, False, False, False, False, False, False,\n",
       "       False, False, False, False, False, False, False, False, False,\n",
       "       False, False, False, False, False, False, False, False, False,\n",
       "       False, False, False, False, False, False, False])"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CO=coverage != -1\n",
    "CO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A7J11_01233+.ctg_cov</th>\n",
       "      <th>aac_3__II-.ctg_cov</th>\n",
       "      <th>aac_6___Ib+.ctg_cov</th>\n",
       "      <th>aadA-.ctg_cov</th>\n",
       "      <th>aadA1+.ctg_cov</th>\n",
       "      <th>aadA2+.ctg_cov</th>\n",
       "      <th>ant_2____Ia.ctg_cov</th>\n",
       "      <th>aphA16.ctg_cov</th>\n",
       "      <th>aph_3___Ia.ctg_cov</th>\n",
       "      <th>aph_3___XV.ctg_cov</th>\n",
       "      <th>...</th>\n",
       "      <th>mph_E_.ctg_cov</th>\n",
       "      <th>msr_E_.ctg_cov</th>\n",
       "      <th>oqxA-.ctg_cov</th>\n",
       "      <th>oqxB-.ctg_cov</th>\n",
       "      <th>qacE-.ctg_cov</th>\n",
       "      <th>qnrS-.ctg_cov</th>\n",
       "      <th>sat2+.ctg_cov</th>\n",
       "      <th>sul1.ctg_cov</th>\n",
       "      <th>sul2.ctg_cov</th>\n",
       "      <th>tet_C_.ctg_cov</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KPN12_S2_L001</th>\n",
       "      <td>51.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>46.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20.5</td>\n",
       "      <td>28.6</td>\n",
       "      <td>20.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN13_S3_L001</th>\n",
       "      <td>37.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24.7</td>\n",
       "      <td>28.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN14_S4_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.7</td>\n",
       "      <td>19.1</td>\n",
       "      <td>17.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN15_S5_L001</th>\n",
       "      <td>82.3</td>\n",
       "      <td>98.8</td>\n",
       "      <td>78.4</td>\n",
       "      <td>57.4</td>\n",
       "      <td>173.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>12.2</td>\n",
       "      <td>15.9</td>\n",
       "      <td>17.9</td>\n",
       "      <td>22.9</td>\n",
       "      <td>52.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24.3</td>\n",
       "      <td>86.9</td>\n",
       "      <td>69.7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNH_S7_L001</th>\n",
       "      <td>35.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.5</td>\n",
       "      <td>23.2</td>\n",
       "      <td>24.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>43.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNI_S8_L001</th>\n",
       "      <td>30.3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.9</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.5</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24.4</td>\n",
       "      <td>27.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1</td>\n",
       "      <td>2.7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNJ_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.6</td>\n",
       "      <td>37.4</td>\n",
       "      <td>29.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNK_S1_L001</th>\n",
       "      <td>111.8</td>\n",
       "      <td>151.5</td>\n",
       "      <td>165.7</td>\n",
       "      <td>141.0</td>\n",
       "      <td>271.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>27.8</td>\n",
       "      <td>38.8</td>\n",
       "      <td>29.7</td>\n",
       "      <td>39.2</td>\n",
       "      <td>94.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.3</td>\n",
       "      <td>166.3</td>\n",
       "      <td>139.6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               A7J11_01233+.ctg_cov  aac_3__II-.ctg_cov  aac_6___Ib+.ctg_cov  \\\n",
       "samples                                                                        \n",
       "KPN12_S2_L001                  51.8                 NaN                 46.5   \n",
       "KPN13_S3_L001                  37.7                 NaN                 41.0   \n",
       "KPN14_S4_L001                   NaN                 NaN                  NaN   \n",
       "KPN15_S5_L001                  82.3                98.8                 78.4   \n",
       "KPNH_S7_L001                   35.9                 NaN                 40.9   \n",
       "KPNI_S8_L001                   30.3                 2.0                 34.0   \n",
       "KPNJ_S2_L001                    NaN                 NaN                  NaN   \n",
       "KPNK_S1_L001                  111.8               151.5                165.7   \n",
       "\n",
       "               aadA-.ctg_cov  aadA1+.ctg_cov  aadA2+.ctg_cov  \\\n",
       "samples                                                        \n",
       "KPN12_S2_L001            NaN             NaN            36.5   \n",
       "KPN13_S3_L001            NaN             NaN             NaN   \n",
       "KPN14_S4_L001            NaN             NaN             NaN   \n",
       "KPN15_S5_L001           57.4           173.0             NaN   \n",
       "KPNH_S7_L001             NaN             NaN            36.6   \n",
       "KPNI_S8_L001             NaN             2.7             NaN   \n",
       "KPNJ_S2_L001             NaN             NaN             NaN   \n",
       "KPNK_S1_L001           141.0           271.4             NaN   \n",
       "\n",
       "               ant_2____Ia.ctg_cov  aphA16.ctg_cov  aph_3___Ia.ctg_cov  \\\n",
       "samples                                                                  \n",
       "KPN12_S2_L001                  NaN             NaN                21.0   \n",
       "KPN13_S3_L001                  NaN             NaN                 NaN   \n",
       "KPN14_S4_L001                  NaN             NaN                17.1   \n",
       "KPN15_S5_L001                  NaN             NaN                 NaN   \n",
       "KPNH_S7_L001                   NaN             NaN                18.6   \n",
       "KPNI_S8_L001                   1.9             2.0                 NaN   \n",
       "KPNJ_S2_L001                   NaN             NaN                40.4   \n",
       "KPNK_S1_L001                   NaN             NaN                 NaN   \n",
       "\n",
       "               aph_3___XV.ctg_cov  ...  mph_E_.ctg_cov  msr_E_.ctg_cov  \\\n",
       "samples                            ...                                   \n",
       "KPN12_S2_L001                 NaN  ...             NaN             NaN   \n",
       "KPN13_S3_L001                19.0  ...             NaN             NaN   \n",
       "KPN14_S4_L001                 NaN  ...             NaN             NaN   \n",
       "KPN15_S5_L001                 NaN  ...            12.2            15.9   \n",
       "KPNH_S7_L001                  NaN  ...             NaN             NaN   \n",
       "KPNI_S8_L001                 15.5  ...             NaN             NaN   \n",
       "KPNJ_S2_L001                  NaN  ...             NaN             NaN   \n",
       "KPNK_S1_L001                  NaN  ...            27.8            38.8   \n",
       "\n",
       "               oqxA-.ctg_cov  oqxB-.ctg_cov  qacE-.ctg_cov  qnrS-.ctg_cov  \\\n",
       "samples                                                                     \n",
       "KPN12_S2_L001           20.5           28.6           20.4            NaN   \n",
       "KPN13_S3_L001           24.7           28.3            NaN           39.2   \n",
       "KPN14_S4_L001           17.7           19.1           17.9            NaN   \n",
       "KPN15_S5_L001           17.9           22.9           52.5            NaN   \n",
       "KPNH_S7_L001            22.5           23.2           24.8            NaN   \n",
       "KPNI_S8_L001            24.4           27.7            NaN           35.0   \n",
       "KPNJ_S2_L001            30.6           37.4           29.9            NaN   \n",
       "KPNK_S1_L001            29.7           39.2           94.7            NaN   \n",
       "\n",
       "               sat2+.ctg_cov  sul1.ctg_cov  sul2.ctg_cov  tet_C_.ctg_cov  \n",
       "samples                                                                   \n",
       "KPN12_S2_L001            NaN          33.6           NaN             NaN  \n",
       "KPN13_S3_L001            NaN           NaN           NaN             NaN  \n",
       "KPN14_S4_L001            NaN          18.4           NaN            19.1  \n",
       "KPN15_S5_L001           24.3          86.9          69.7             NaN  \n",
       "KPNH_S7_L001             NaN          43.8           NaN             NaN  \n",
       "KPNI_S8_L001             NaN           2.1           2.7             NaN  \n",
       "KPNJ_S2_L001             NaN          38.6           NaN            44.6  \n",
       "KPNK_S1_L001            29.3         166.3         139.6             NaN  \n",
       "\n",
       "[8 rows x 49 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cover=datos.iloc[:,CO]\n",
    "cover ###----> it shows the locus that have .ctg_cov suffix with a value (if it has a value on it)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A7J11_01233+.ctg_cov</th>\n",
       "      <th>aac_3__II-.ctg_cov</th>\n",
       "      <th>aac_6___Ib+.ctg_cov</th>\n",
       "      <th>aadA-.ctg_cov</th>\n",
       "      <th>aadA1+.ctg_cov</th>\n",
       "      <th>aadA2+.ctg_cov</th>\n",
       "      <th>ant_2____Ia.ctg_cov</th>\n",
       "      <th>aphA16.ctg_cov</th>\n",
       "      <th>aph_3___Ia.ctg_cov</th>\n",
       "      <th>aph_3___XV.ctg_cov</th>\n",
       "      <th>...</th>\n",
       "      <th>mph_E_.ctg_cov</th>\n",
       "      <th>msr_E_.ctg_cov</th>\n",
       "      <th>oqxA-.ctg_cov</th>\n",
       "      <th>oqxB-.ctg_cov</th>\n",
       "      <th>qacE-.ctg_cov</th>\n",
       "      <th>qnrS-.ctg_cov</th>\n",
       "      <th>sat2+.ctg_cov</th>\n",
       "      <th>sul1.ctg_cov</th>\n",
       "      <th>sul2.ctg_cov</th>\n",
       "      <th>tet_C_.ctg_cov</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KPN12_S2_L001</th>\n",
       "      <td>51.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>46.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20.5</td>\n",
       "      <td>28.6</td>\n",
       "      <td>20.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN13_S3_L001</th>\n",
       "      <td>37.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24.7</td>\n",
       "      <td>28.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN14_S4_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.7</td>\n",
       "      <td>19.1</td>\n",
       "      <td>17.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN15_S5_L001</th>\n",
       "      <td>82.3</td>\n",
       "      <td>98.8</td>\n",
       "      <td>78.4</td>\n",
       "      <td>57.4</td>\n",
       "      <td>173.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>12.2</td>\n",
       "      <td>15.9</td>\n",
       "      <td>17.9</td>\n",
       "      <td>22.9</td>\n",
       "      <td>52.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24.3</td>\n",
       "      <td>86.9</td>\n",
       "      <td>69.7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNH_S7_L001</th>\n",
       "      <td>35.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.5</td>\n",
       "      <td>23.2</td>\n",
       "      <td>24.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>43.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNI_S8_L001</th>\n",
       "      <td>30.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.5</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24.4</td>\n",
       "      <td>27.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNJ_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.6</td>\n",
       "      <td>37.4</td>\n",
       "      <td>29.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNK_S1_L001</th>\n",
       "      <td>111.8</td>\n",
       "      <td>151.5</td>\n",
       "      <td>165.7</td>\n",
       "      <td>141.0</td>\n",
       "      <td>271.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>27.8</td>\n",
       "      <td>38.8</td>\n",
       "      <td>29.7</td>\n",
       "      <td>39.2</td>\n",
       "      <td>94.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.3</td>\n",
       "      <td>166.3</td>\n",
       "      <td>139.6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               A7J11_01233+.ctg_cov  aac_3__II-.ctg_cov  aac_6___Ib+.ctg_cov  \\\n",
       "samples                                                                        \n",
       "KPN12_S2_L001                  51.8                 NaN                 46.5   \n",
       "KPN13_S3_L001                  37.7                 NaN                 41.0   \n",
       "KPN14_S4_L001                   NaN                 NaN                  NaN   \n",
       "KPN15_S5_L001                  82.3                98.8                 78.4   \n",
       "KPNH_S7_L001                   35.9                 NaN                 40.9   \n",
       "KPNI_S8_L001                   30.3                 NaN                 34.0   \n",
       "KPNJ_S2_L001                    NaN                 NaN                  NaN   \n",
       "KPNK_S1_L001                  111.8               151.5                165.7   \n",
       "\n",
       "               aadA-.ctg_cov  aadA1+.ctg_cov  aadA2+.ctg_cov  \\\n",
       "samples                                                        \n",
       "KPN12_S2_L001            NaN             NaN            36.5   \n",
       "KPN13_S3_L001            NaN             NaN             NaN   \n",
       "KPN14_S4_L001            NaN             NaN             NaN   \n",
       "KPN15_S5_L001           57.4           173.0             NaN   \n",
       "KPNH_S7_L001             NaN             NaN            36.6   \n",
       "KPNI_S8_L001             NaN             NaN             NaN   \n",
       "KPNJ_S2_L001             NaN             NaN             NaN   \n",
       "KPNK_S1_L001           141.0           271.4             NaN   \n",
       "\n",
       "               ant_2____Ia.ctg_cov  aphA16.ctg_cov  aph_3___Ia.ctg_cov  \\\n",
       "samples                                                                  \n",
       "KPN12_S2_L001                  NaN             NaN                21.0   \n",
       "KPN13_S3_L001                  NaN             NaN                 NaN   \n",
       "KPN14_S4_L001                  NaN             NaN                17.1   \n",
       "KPN15_S5_L001                  NaN             NaN                 NaN   \n",
       "KPNH_S7_L001                   NaN             NaN                18.6   \n",
       "KPNI_S8_L001                   NaN             NaN                 NaN   \n",
       "KPNJ_S2_L001                   NaN             NaN                40.4   \n",
       "KPNK_S1_L001                   NaN             NaN                 NaN   \n",
       "\n",
       "               aph_3___XV.ctg_cov  ...  mph_E_.ctg_cov  msr_E_.ctg_cov  \\\n",
       "samples                            ...                                   \n",
       "KPN12_S2_L001                 NaN  ...             NaN             NaN   \n",
       "KPN13_S3_L001                19.0  ...             NaN             NaN   \n",
       "KPN14_S4_L001                 NaN  ...             NaN             NaN   \n",
       "KPN15_S5_L001                 NaN  ...            12.2            15.9   \n",
       "KPNH_S7_L001                  NaN  ...             NaN             NaN   \n",
       "KPNI_S8_L001                 15.5  ...             NaN             NaN   \n",
       "KPNJ_S2_L001                  NaN  ...             NaN             NaN   \n",
       "KPNK_S1_L001                  NaN  ...            27.8            38.8   \n",
       "\n",
       "               oqxA-.ctg_cov  oqxB-.ctg_cov  qacE-.ctg_cov  qnrS-.ctg_cov  \\\n",
       "samples                                                                     \n",
       "KPN12_S2_L001           20.5           28.6           20.4            NaN   \n",
       "KPN13_S3_L001           24.7           28.3            NaN           39.2   \n",
       "KPN14_S4_L001           17.7           19.1           17.9            NaN   \n",
       "KPN15_S5_L001           17.9           22.9           52.5            NaN   \n",
       "KPNH_S7_L001            22.5           23.2           24.8            NaN   \n",
       "KPNI_S8_L001            24.4           27.7            NaN           35.0   \n",
       "KPNJ_S2_L001            30.6           37.4           29.9            NaN   \n",
       "KPNK_S1_L001            29.7           39.2           94.7            NaN   \n",
       "\n",
       "               sat2+.ctg_cov  sul1.ctg_cov  sul2.ctg_cov  tet_C_.ctg_cov  \n",
       "samples                                                                   \n",
       "KPN12_S2_L001            NaN          33.6           NaN             NaN  \n",
       "KPN13_S3_L001            NaN           NaN           NaN             NaN  \n",
       "KPN14_S4_L001            NaN          18.4           NaN            19.1  \n",
       "KPN15_S5_L001           24.3          86.9          69.7             NaN  \n",
       "KPNH_S7_L001             NaN          43.8           NaN             NaN  \n",
       "KPNI_S8_L001             NaN           NaN           NaN             NaN  \n",
       "KPNJ_S2_L001             NaN          38.6           NaN            44.6  \n",
       "KPNK_S1_L001            29.3         166.3         139.6             NaN  \n",
       "\n",
       "[8 rows x 49 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cover_filtrated=cover[cover>10]  ###----> sort the values coverage higher than 10 \n",
    "cover_filtrated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "samples                            \n",
       "KPN12_S2_L001  A7J11_01233+.ctg_cov     51.8\n",
       "               aac_6___Ib+.ctg_cov      46.5\n",
       "               aadA2+.ctg_cov           36.5\n",
       "               aph_3___Ia.ctg_cov       21.0\n",
       "               bla-_6.ctg_cov           44.6\n",
       "                                       ...  \n",
       "KPNK_S1_L001   oqxB-.ctg_cov            39.2\n",
       "               qacE-.ctg_cov            94.7\n",
       "               sat2+.ctg_cov            29.3\n",
       "               sul1.ctg_cov            166.3\n",
       "               sul2.ctg_cov            139.6\n",
       "Length: 146, dtype: float64"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_coverage=cover_filtrated.stack()\n",
    "final_coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>A7J11_01233+.ctg_cov</td>\n",
       "      <td>51.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aac_6___Ib+.ctg_cov</td>\n",
       "      <td>46.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aadA2+.ctg_cov</td>\n",
       "      <td>36.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aph_3___Ia.ctg_cov</td>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>bla-_6.ctg_cov</td>\n",
       "      <td>44.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>oqxB-.ctg_cov</td>\n",
       "      <td>39.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>qacE-.ctg_cov</td>\n",
       "      <td>94.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sat2+.ctg_cov</td>\n",
       "      <td>29.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sul1.ctg_cov</td>\n",
       "      <td>166.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sul2.ctg_cov</td>\n",
       "      <td>139.6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>146 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           samples               level_1      0\n",
       "0    KPN12_S2_L001  A7J11_01233+.ctg_cov   51.8\n",
       "1    KPN12_S2_L001   aac_6___Ib+.ctg_cov   46.5\n",
       "2    KPN12_S2_L001        aadA2+.ctg_cov   36.5\n",
       "3    KPN12_S2_L001    aph_3___Ia.ctg_cov   21.0\n",
       "4    KPN12_S2_L001        bla-_6.ctg_cov   44.6\n",
       "..             ...                   ...    ...\n",
       "141   KPNK_S1_L001         oqxB-.ctg_cov   39.2\n",
       "142   KPNK_S1_L001         qacE-.ctg_cov   94.7\n",
       "143   KPNK_S1_L001         sat2+.ctg_cov   29.3\n",
       "144   KPNK_S1_L001          sul1.ctg_cov  166.3\n",
       "145   KPNK_S1_L001          sul2.ctg_cov  139.6\n",
       "\n",
       "[146 rows x 3 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_coverage_df = final_coverage.reset_index(level=[0,1])\n",
    "final_coverage_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>level_1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A7J11_01233+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>aac_6___Ib+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>aadA2+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>aph_3___Ia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bla-_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>oqxB-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>qacE-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>sat2+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>sul1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>sul2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>146 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          level_1\n",
       "0    A7J11_01233+\n",
       "1     aac_6___Ib+\n",
       "2          aadA2+\n",
       "3      aph_3___Ia\n",
       "4          bla-_6\n",
       "..            ...\n",
       "141         oqxB-\n",
       "142         qacE-\n",
       "143         sat2+\n",
       "144          sul1\n",
       "145          sul2\n",
       "\n",
       "[146 rows x 1 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fixed_level1_cover=final_coverage_df.level_1.str.rstrip('.ctg_cov')\n",
    "fixed_level1_cover=pd.DataFrame(fixed_level1_cover)\n",
    "fixed_level1_cover"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>A7J11_01233+</td>\n",
       "      <td>51.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aac_6___Ib+</td>\n",
       "      <td>46.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aadA2+</td>\n",
       "      <td>36.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aph_3___Ia</td>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>bla-_6</td>\n",
       "      <td>44.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>oqxB-</td>\n",
       "      <td>39.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>qacE-</td>\n",
       "      <td>94.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sat2+</td>\n",
       "      <td>29.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sul1</td>\n",
       "      <td>166.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sul2</td>\n",
       "      <td>139.6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>146 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           samples       level_1      0\n",
       "0    KPN12_S2_L001  A7J11_01233+   51.8\n",
       "1    KPN12_S2_L001   aac_6___Ib+   46.5\n",
       "2    KPN12_S2_L001        aadA2+   36.5\n",
       "3    KPN12_S2_L001    aph_3___Ia   21.0\n",
       "4    KPN12_S2_L001        bla-_6   44.6\n",
       "..             ...           ...    ...\n",
       "141   KPNK_S1_L001         oqxB-   39.2\n",
       "142   KPNK_S1_L001         qacE-   94.7\n",
       "143   KPNK_S1_L001         sat2+   29.3\n",
       "144   KPNK_S1_L001          sul1  166.3\n",
       "145   KPNK_S1_L001          sul2  139.6\n",
       "\n",
       "[146 rows x 3 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_coverage_df['level_1']=fixed_level1_cover['level_1']\n",
    "final_coverage_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>coverage</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPN12_S2_L001</th>\n",
       "      <th>A7J11_01233+</th>\n",
       "      <td>51.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aac_6___Ib+</th>\n",
       "      <td>46.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aadA2+</th>\n",
       "      <td>36.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aph_3___Ia</th>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bla-_6</th>\n",
       "      <td>44.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPNK_S1_L001</th>\n",
       "      <th>oqxB-</th>\n",
       "      <td>39.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>qacE-</th>\n",
       "      <td>94.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sat2+</th>\n",
       "      <td>29.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul1</th>\n",
       "      <td>166.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul2</th>\n",
       "      <td>139.6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>146 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            coverage\n",
       "samples       level_1               \n",
       "KPN12_S2_L001 A7J11_01233+      51.8\n",
       "              aac_6___Ib+       46.5\n",
       "              aadA2+            36.5\n",
       "              aph_3___Ia        21.0\n",
       "              bla-_6            44.6\n",
       "...                              ...\n",
       "KPNK_S1_L001  oqxB-             39.2\n",
       "              qacE-             94.7\n",
       "              sat2+             29.3\n",
       "              sul1             166.3\n",
       "              sul2             139.6\n",
       "\n",
       "[146 rows x 1 columns]"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#####----------------------------------> data frame 2 ---> coverage arreglados!!\n",
    "final_coverage_fixed=final_coverage_df.set_index(['samples', 'level_1'])\n",
    "final_coverage_fixed=final_coverage_fixed.rename(columns = {0:'coverage'}, inplace = False)\n",
    "final_coverage_fixed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#allele</th>\n",
       "      <th>gene family</th>\n",
       "      <th>refseq nucleotide</th>\n",
       "      <th>class</th>\n",
       "      <th>subclass</th>\n",
       "      <th>refseq protein</th>\n",
       "      <th>genbank protein</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(2')-IIa</td>\n",
       "      <td>NG_047225.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>KASUGAMYCIN</td>\n",
       "      <td>WP_063839881.1</td>\n",
       "      <td>BAM16262.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(2')-IIb</td>\n",
       "      <td>NG_055672.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>KASUGAMYCIN</td>\n",
       "      <td>WP_071224044.1</td>\n",
       "      <td>APB03221.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(2')-Ia</td>\n",
       "      <td>NG_047226.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>GENTAMICIN/TOBRAMCYIN</td>\n",
       "      <td>WP_004918308.1</td>\n",
       "      <td>AAA03550.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(2')-Ib</td>\n",
       "      <td>NG_047227.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>GENTAMICIN/TOBRAMCYIN</td>\n",
       "      <td>WP_003881640.1</td>\n",
       "      <td>AAC44793.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(2')-Ic</td>\n",
       "      <td>NG_047229.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>GENTAMICIN/TOBRAMCYIN</td>\n",
       "      <td>WP_003899880.1</td>\n",
       "      <td>AAB17563.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7239</th>\n",
       "      <td>uhpT_W425R</td>\n",
       "      <td>uhpT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FOSFOMYCIN</td>\n",
       "      <td>FOSFOMYCIN</td>\n",
       "      <td>WP_001008722.1</td>\n",
       "      <td>CAG39240.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7240</th>\n",
       "      <td>walK_L10F</td>\n",
       "      <td>walK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIPOPEPTIDE</td>\n",
       "      <td>DAPTOMYCIN</td>\n",
       "      <td>WP_000871610.1</td>\n",
       "      <td>CAG39047.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7241</th>\n",
       "      <td>walK_S221P</td>\n",
       "      <td>walK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIPOPEPTIDE</td>\n",
       "      <td>DAPTOMYCIN</td>\n",
       "      <td>WP_000871610.1</td>\n",
       "      <td>CAG39047.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7242</th>\n",
       "      <td>walK_R263C</td>\n",
       "      <td>walK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIPOPEPTIDE</td>\n",
       "      <td>DAPTOMYCIN</td>\n",
       "      <td>WP_000871610.1</td>\n",
       "      <td>CAG39047.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7243</th>\n",
       "      <td>yybT_I440S</td>\n",
       "      <td>yybT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIPOPEPTIDE</td>\n",
       "      <td>DAPTOMYCIN</td>\n",
       "      <td>WP_002356022.1</td>\n",
       "      <td>AAO79895.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7244 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #allele  gene family refseq nucleotide           class  \\\n",
       "0            NaN  aac(2')-IIa       NG_047225.1  AMINOGLYCOSIDE   \n",
       "1            NaN  aac(2')-IIb       NG_055672.1  AMINOGLYCOSIDE   \n",
       "2            NaN   aac(2')-Ia       NG_047226.1  AMINOGLYCOSIDE   \n",
       "3            NaN   aac(2')-Ib       NG_047227.1  AMINOGLYCOSIDE   \n",
       "4            NaN   aac(2')-Ic       NG_047229.1  AMINOGLYCOSIDE   \n",
       "...          ...          ...               ...             ...   \n",
       "7239  uhpT_W425R         uhpT               NaN      FOSFOMYCIN   \n",
       "7240   walK_L10F         walK               NaN     LIPOPEPTIDE   \n",
       "7241  walK_S221P         walK               NaN     LIPOPEPTIDE   \n",
       "7242  walK_R263C         walK               NaN     LIPOPEPTIDE   \n",
       "7243  yybT_I440S         yybT               NaN     LIPOPEPTIDE   \n",
       "\n",
       "                   subclass  refseq protein genbank protein  \n",
       "0               KASUGAMYCIN  WP_063839881.1      BAM16262.1  \n",
       "1               KASUGAMYCIN  WP_071224044.1      APB03221.1  \n",
       "2     GENTAMICIN/TOBRAMCYIN  WP_004918308.1      AAA03550.1  \n",
       "3     GENTAMICIN/TOBRAMCYIN  WP_003881640.1      AAC44793.1  \n",
       "4     GENTAMICIN/TOBRAMCYIN  WP_003899880.1      AAB17563.1  \n",
       "...                     ...             ...             ...  \n",
       "7239             FOSFOMYCIN  WP_001008722.1      CAG39240.1  \n",
       "7240             DAPTOMYCIN  WP_000871610.1      CAG39047.1  \n",
       "7241             DAPTOMYCIN  WP_000871610.1      CAG39047.1  \n",
       "7242             DAPTOMYCIN  WP_000871610.1      CAG39047.1  \n",
       "7243             DAPTOMYCIN  WP_002356022.1      AAO79895.1  \n",
       "\n",
       "[7244 rows x 7 columns]"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "database_anthony=pd.read_csv(\"/Users/felipedb/Documents/data/projects/training/taller_agrosavia/ariba_exercise/exercise_2_ncbi_acquired_genes_metadata.csv\", sep=\",\")\n",
    "database_anthony"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#allele</th>\n",
       "      <th>gene family</th>\n",
       "      <th>refseq nucleotide</th>\n",
       "      <th>class</th>\n",
       "      <th>subclass</th>\n",
       "      <th>refseq protein</th>\n",
       "      <th>genbank protein</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(2')-IIa</td>\n",
       "      <td>NG_047225.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>KASUGAMYCIN</td>\n",
       "      <td>WP_063839881.1</td>\n",
       "      <td>BAM16262.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(2')-IIb</td>\n",
       "      <td>NG_055672.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>KASUGAMYCIN</td>\n",
       "      <td>WP_071224044.1</td>\n",
       "      <td>APB03221.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(2')-Ia</td>\n",
       "      <td>NG_047226.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>GENTAMICIN/TOBRAMCYIN</td>\n",
       "      <td>WP_004918308.1</td>\n",
       "      <td>AAA03550.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(2')-Ib</td>\n",
       "      <td>NG_047227.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>GENTAMICIN/TOBRAMCYIN</td>\n",
       "      <td>WP_003881640.1</td>\n",
       "      <td>AAC44793.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(2')-Ic</td>\n",
       "      <td>NG_047229.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>GENTAMICIN/TOBRAMCYIN</td>\n",
       "      <td>WP_003899880.1</td>\n",
       "      <td>AAB17563.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7239</th>\n",
       "      <td>uhpT_W425R</td>\n",
       "      <td>uhpT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FOSFOMYCIN</td>\n",
       "      <td>FOSFOMYCIN</td>\n",
       "      <td>WP_001008722.1</td>\n",
       "      <td>CAG39240.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7240</th>\n",
       "      <td>walK_L10F</td>\n",
       "      <td>walK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIPOPEPTIDE</td>\n",
       "      <td>DAPTOMYCIN</td>\n",
       "      <td>WP_000871610.1</td>\n",
       "      <td>CAG39047.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7241</th>\n",
       "      <td>walK_S221P</td>\n",
       "      <td>walK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIPOPEPTIDE</td>\n",
       "      <td>DAPTOMYCIN</td>\n",
       "      <td>WP_000871610.1</td>\n",
       "      <td>CAG39047.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7242</th>\n",
       "      <td>walK_R263C</td>\n",
       "      <td>walK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIPOPEPTIDE</td>\n",
       "      <td>DAPTOMYCIN</td>\n",
       "      <td>WP_000871610.1</td>\n",
       "      <td>CAG39047.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7243</th>\n",
       "      <td>yybT_I440S</td>\n",
       "      <td>yybT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIPOPEPTIDE</td>\n",
       "      <td>DAPTOMYCIN</td>\n",
       "      <td>WP_002356022.1</td>\n",
       "      <td>AAO79895.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7170 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #allele  gene family refseq nucleotide           class  \\\n",
       "0            NaN  aac(2')-IIa       NG_047225.1  AMINOGLYCOSIDE   \n",
       "1            NaN  aac(2')-IIb       NG_055672.1  AMINOGLYCOSIDE   \n",
       "2            NaN   aac(2')-Ia       NG_047226.1  AMINOGLYCOSIDE   \n",
       "3            NaN   aac(2')-Ib       NG_047227.1  AMINOGLYCOSIDE   \n",
       "4            NaN   aac(2')-Ic       NG_047229.1  AMINOGLYCOSIDE   \n",
       "...          ...          ...               ...             ...   \n",
       "7239  uhpT_W425R         uhpT               NaN      FOSFOMYCIN   \n",
       "7240   walK_L10F         walK               NaN     LIPOPEPTIDE   \n",
       "7241  walK_S221P         walK               NaN     LIPOPEPTIDE   \n",
       "7242  walK_R263C         walK               NaN     LIPOPEPTIDE   \n",
       "7243  yybT_I440S         yybT               NaN     LIPOPEPTIDE   \n",
       "\n",
       "                   subclass  refseq protein genbank protein  \n",
       "0               KASUGAMYCIN  WP_063839881.1      BAM16262.1  \n",
       "1               KASUGAMYCIN  WP_071224044.1      APB03221.1  \n",
       "2     GENTAMICIN/TOBRAMCYIN  WP_004918308.1      AAA03550.1  \n",
       "3     GENTAMICIN/TOBRAMCYIN  WP_003881640.1      AAC44793.1  \n",
       "4     GENTAMICIN/TOBRAMCYIN  WP_003899880.1      AAB17563.1  \n",
       "...                     ...             ...             ...  \n",
       "7239             FOSFOMYCIN  WP_001008722.1      CAG39240.1  \n",
       "7240             DAPTOMYCIN  WP_000871610.1      CAG39047.1  \n",
       "7241             DAPTOMYCIN  WP_000871610.1      CAG39047.1  \n",
       "7242             DAPTOMYCIN  WP_000871610.1      CAG39047.1  \n",
       "7243             DAPTOMYCIN  WP_002356022.1      AAO79895.1  \n",
       "\n",
       "[7170 rows x 7 columns]"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NCBI_filtered = database_anthony[database_anthony[\"refseq nucleotide\"].str.find(\"NG_\")!=-1]\n",
    "NCBI_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Int64Index([-1, 12, -1, -1, -1, 10, -1, -1, -1, 11,\n",
       "            ...\n",
       "            -1, -1, -1, -1, -1, -1, -1, -1, -1, -1],\n",
       "           dtype='int64', length=232)"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_seq=datos.columns.str.find(\".ref_seq\")###----------------->reference sequence filtering from my samples\n",
    "ref_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False,  True,\n",
       "       False, False, False,  True, False, False, False,  True, False,\n",
       "       False, False,  True, False, False, False,  True, False, False,\n",
       "       False,  True, False, False, False,  True, False, False, False,\n",
       "        True, False, False, False,  True, False, False, False, False,\n",
       "       False, False, False, False, False, False, False, False, False,\n",
       "       False, False, False, False, False, False, False, False, False,\n",
       "       False, False, False, False, False, False, False, False, False,\n",
       "       False, False, False, False, False, False, False])"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RF=ref_seq != -1\n",
    "RF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A7J11_01233+.ref_seq</th>\n",
       "      <th>aac_3__II-.ref_seq</th>\n",
       "      <th>aac_6___Ib+.ref_seq</th>\n",
       "      <th>aadA-.ref_seq</th>\n",
       "      <th>aadA1+.ref_seq</th>\n",
       "      <th>aadA2+.ref_seq</th>\n",
       "      <th>ant_2____Ia.ref_seq</th>\n",
       "      <th>aphA16.ref_seq</th>\n",
       "      <th>aph_3___Ia.ref_seq</th>\n",
       "      <th>aph_3___XV.ref_seq</th>\n",
       "      <th>...</th>\n",
       "      <th>mph_E_.ref_seq</th>\n",
       "      <th>msr_E_.ref_seq</th>\n",
       "      <th>oqxA-.ref_seq</th>\n",
       "      <th>oqxB-.ref_seq</th>\n",
       "      <th>qacE-.ref_seq</th>\n",
       "      <th>qnrS-.ref_seq</th>\n",
       "      <th>sat2+.ref_seq</th>\n",
       "      <th>sul1.ref_seq</th>\n",
       "      <th>sul2.ref_seq</th>\n",
       "      <th>tet_C_.ref_seq</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KPN12_S2_L001</th>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aac_6___Ib.NG_056043.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aadA2.NG_051846.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aph_3___Ia.NG_047430.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>oqxA.NG_048024.1</td>\n",
       "      <td>oqxB.NG_048025.1</td>\n",
       "      <td>qacEdelta1.NG_048042.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sul1.NG_048082.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN13_S3_L001</th>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aac_6___Ib4.NG_051844.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aph_3___XV.NG_052510.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>oqxA5.NG_050423.1</td>\n",
       "      <td>oqxB18.NG_050436.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>qnrS1.NG_050543.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN14_S4_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aph_3___Ia.NG_047430.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>oqxA10.NG_050418.1</td>\n",
       "      <td>oqxB22.NG_050441.1</td>\n",
       "      <td>qacEdelta1.NG_048042.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sul1.NG_048082.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tet_C_.NG_048177.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPN15_S5_L001</th>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>aac_3__IId.NG_047251.1</td>\n",
       "      <td>aac_6___Ib.NG_056043.1</td>\n",
       "      <td>aadA13.NG_052310.1</td>\n",
       "      <td>aadA1.NG_052536.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>mph_E_.NG_064660.1</td>\n",
       "      <td>msr_E_.NG_048007.1</td>\n",
       "      <td>oqxA6.NG_050424.1</td>\n",
       "      <td>oqxB20.NG_050439.1</td>\n",
       "      <td>qacEdelta1.NG_048042.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sat2.NG_048068.1</td>\n",
       "      <td>sul1.NG_048081.1</td>\n",
       "      <td>sul2.NG_051852.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNH_S7_L001</th>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aac_6___Ib.NG_056043.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aadA2.NG_047343.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aph_3___Ia.NG_047430.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>oqxA.NG_048024.1</td>\n",
       "      <td>oqxB.NG_048025.1</td>\n",
       "      <td>qacEdelta1.NG_048042.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sul1.NG_048082.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNI_S8_L001</th>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>aac_3__IId.NG_047251.1</td>\n",
       "      <td>aac_6___Ib4.NG_051844.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aadA8.NG_052405.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ant_2____Ia.NG_052437.1</td>\n",
       "      <td>aphA16.NG_051908.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aph_3___XV.NG_052510.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>oqxA5.NG_050423.1</td>\n",
       "      <td>oqxB18.NG_050436.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>qnrS1.NG_050543.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sul1.NG_048089.1</td>\n",
       "      <td>sul2.NG_051852.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNJ_S2_L001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aph_3___Ia.NG_047430.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>oqxA10.NG_050418.1</td>\n",
       "      <td>oqxB22.NG_050441.1</td>\n",
       "      <td>qacEdelta1.NG_048042.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sul1.NG_048082.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tet_C_.NG_048177.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KPNK_S1_L001</th>\n",
       "      <td>A7J11_01233.NG_047213.1</td>\n",
       "      <td>aac_3__IId.NG_047251.1</td>\n",
       "      <td>aac_6___Ib.NG_056043.1</td>\n",
       "      <td>aadA13.NG_052310.1</td>\n",
       "      <td>aadA1.NG_052536.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>mph_E_.NG_064660.1</td>\n",
       "      <td>msr_E_.NG_048007.1</td>\n",
       "      <td>oqxA6.NG_050424.1</td>\n",
       "      <td>oqxB20.NG_050439.1</td>\n",
       "      <td>qacEdelta1.NG_048042.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sat2.NG_048068.1</td>\n",
       "      <td>sul1.NG_048081.1</td>\n",
       "      <td>sul2.NG_051852.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  A7J11_01233+.ref_seq      aac_3__II-.ref_seq  \\\n",
       "samples                                                          \n",
       "KPN12_S2_L001  A7J11_01233.NG_047213.1                     NaN   \n",
       "KPN13_S3_L001  A7J11_01233.NG_047213.1                     NaN   \n",
       "KPN14_S4_L001                      NaN                     NaN   \n",
       "KPN15_S5_L001  A7J11_01233.NG_047213.1  aac_3__IId.NG_047251.1   \n",
       "KPNH_S7_L001   A7J11_01233.NG_047213.1                     NaN   \n",
       "KPNI_S8_L001   A7J11_01233.NG_047213.1  aac_3__IId.NG_047251.1   \n",
       "KPNJ_S2_L001                       NaN                     NaN   \n",
       "KPNK_S1_L001   A7J11_01233.NG_047213.1  aac_3__IId.NG_047251.1   \n",
       "\n",
       "                   aac_6___Ib+.ref_seq       aadA-.ref_seq     aadA1+.ref_seq  \\\n",
       "samples                                                                         \n",
       "KPN12_S2_L001   aac_6___Ib.NG_056043.1                 NaN                NaN   \n",
       "KPN13_S3_L001  aac_6___Ib4.NG_051844.1                 NaN                NaN   \n",
       "KPN14_S4_L001                      NaN                 NaN                NaN   \n",
       "KPN15_S5_L001   aac_6___Ib.NG_056043.1  aadA13.NG_052310.1  aadA1.NG_052536.1   \n",
       "KPNH_S7_L001    aac_6___Ib.NG_056043.1                 NaN                NaN   \n",
       "KPNI_S8_L001   aac_6___Ib4.NG_051844.1                 NaN  aadA8.NG_052405.1   \n",
       "KPNJ_S2_L001                       NaN                 NaN                NaN   \n",
       "KPNK_S1_L001    aac_6___Ib.NG_056043.1  aadA13.NG_052310.1  aadA1.NG_052536.1   \n",
       "\n",
       "                  aadA2+.ref_seq      ant_2____Ia.ref_seq      aphA16.ref_seq  \\\n",
       "samples                                                                         \n",
       "KPN12_S2_L001  aadA2.NG_051846.1                      NaN                 NaN   \n",
       "KPN13_S3_L001                NaN                      NaN                 NaN   \n",
       "KPN14_S4_L001                NaN                      NaN                 NaN   \n",
       "KPN15_S5_L001                NaN                      NaN                 NaN   \n",
       "KPNH_S7_L001   aadA2.NG_047343.1                      NaN                 NaN   \n",
       "KPNI_S8_L001                 NaN  ant_2____Ia.NG_052437.1  aphA16.NG_051908.1   \n",
       "KPNJ_S2_L001                 NaN                      NaN                 NaN   \n",
       "KPNK_S1_L001                 NaN                      NaN                 NaN   \n",
       "\n",
       "                   aph_3___Ia.ref_seq      aph_3___XV.ref_seq  ...  \\\n",
       "samples                                                        ...   \n",
       "KPN12_S2_L001  aph_3___Ia.NG_047430.1                     NaN  ...   \n",
       "KPN13_S3_L001                     NaN  aph_3___XV.NG_052510.1  ...   \n",
       "KPN14_S4_L001  aph_3___Ia.NG_047430.1                     NaN  ...   \n",
       "KPN15_S5_L001                     NaN                     NaN  ...   \n",
       "KPNH_S7_L001   aph_3___Ia.NG_047430.1                     NaN  ...   \n",
       "KPNI_S8_L001                      NaN  aph_3___XV.NG_052510.1  ...   \n",
       "KPNJ_S2_L001   aph_3___Ia.NG_047430.1                     NaN  ...   \n",
       "KPNK_S1_L001                      NaN                     NaN  ...   \n",
       "\n",
       "                   mph_E_.ref_seq      msr_E_.ref_seq       oqxA-.ref_seq  \\\n",
       "samples                                                                     \n",
       "KPN12_S2_L001                 NaN                 NaN    oqxA.NG_048024.1   \n",
       "KPN13_S3_L001                 NaN                 NaN   oqxA5.NG_050423.1   \n",
       "KPN14_S4_L001                 NaN                 NaN  oqxA10.NG_050418.1   \n",
       "KPN15_S5_L001  mph_E_.NG_064660.1  msr_E_.NG_048007.1   oqxA6.NG_050424.1   \n",
       "KPNH_S7_L001                  NaN                 NaN    oqxA.NG_048024.1   \n",
       "KPNI_S8_L001                  NaN                 NaN   oqxA5.NG_050423.1   \n",
       "KPNJ_S2_L001                  NaN                 NaN  oqxA10.NG_050418.1   \n",
       "KPNK_S1_L001   mph_E_.NG_064660.1  msr_E_.NG_048007.1   oqxA6.NG_050424.1   \n",
       "\n",
       "                    oqxB-.ref_seq           qacE-.ref_seq      qnrS-.ref_seq  \\\n",
       "samples                                                                        \n",
       "KPN12_S2_L001    oqxB.NG_048025.1  qacEdelta1.NG_048042.1                NaN   \n",
       "KPN13_S3_L001  oqxB18.NG_050436.1                     NaN  qnrS1.NG_050543.1   \n",
       "KPN14_S4_L001  oqxB22.NG_050441.1  qacEdelta1.NG_048042.1                NaN   \n",
       "KPN15_S5_L001  oqxB20.NG_050439.1  qacEdelta1.NG_048042.1                NaN   \n",
       "KPNH_S7_L001     oqxB.NG_048025.1  qacEdelta1.NG_048042.1                NaN   \n",
       "KPNI_S8_L001   oqxB18.NG_050436.1                     NaN  qnrS1.NG_050543.1   \n",
       "KPNJ_S2_L001   oqxB22.NG_050441.1  qacEdelta1.NG_048042.1                NaN   \n",
       "KPNK_S1_L001   oqxB20.NG_050439.1  qacEdelta1.NG_048042.1                NaN   \n",
       "\n",
       "                  sat2+.ref_seq      sul1.ref_seq      sul2.ref_seq  \\\n",
       "samples                                                               \n",
       "KPN12_S2_L001               NaN  sul1.NG_048082.1               NaN   \n",
       "KPN13_S3_L001               NaN               NaN               NaN   \n",
       "KPN14_S4_L001               NaN  sul1.NG_048082.1               NaN   \n",
       "KPN15_S5_L001  sat2.NG_048068.1  sul1.NG_048081.1  sul2.NG_051852.1   \n",
       "KPNH_S7_L001                NaN  sul1.NG_048082.1               NaN   \n",
       "KPNI_S8_L001                NaN  sul1.NG_048089.1  sul2.NG_051852.1   \n",
       "KPNJ_S2_L001                NaN  sul1.NG_048082.1               NaN   \n",
       "KPNK_S1_L001   sat2.NG_048068.1  sul1.NG_048081.1  sul2.NG_051852.1   \n",
       "\n",
       "                   tet_C_.ref_seq  \n",
       "samples                            \n",
       "KPN12_S2_L001                 NaN  \n",
       "KPN13_S3_L001                 NaN  \n",
       "KPN14_S4_L001  tet_C_.NG_048177.1  \n",
       "KPN15_S5_L001                 NaN  \n",
       "KPNH_S7_L001                  NaN  \n",
       "KPNI_S8_L001                  NaN  \n",
       "KPNJ_S2_L001   tet_C_.NG_048177.1  \n",
       "KPNK_S1_L001                  NaN  \n",
       "\n",
       "[8 rows x 49 columns]"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "referenced=datos.iloc[:,RF]\n",
    "referenced"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "samples                            \n",
       "KPN12_S2_L001  A7J11_01233+.ref_seq    A7J11_01233.NG_047213.1\n",
       "               aac_6___Ib+.ref_seq      aac_6___Ib.NG_056043.1\n",
       "               aadA2+.ref_seq                aadA2.NG_051846.1\n",
       "               aph_3___Ia.ref_seq       aph_3___Ia.NG_047430.1\n",
       "               bla-_6.ref_seq           blaSHV_182.NG_050049.1\n",
       "                                                ...           \n",
       "KPNK_S1_L001   oqxB-.ref_seq                oqxB20.NG_050439.1\n",
       "               qacE-.ref_seq            qacEdelta1.NG_048042.1\n",
       "               sat2+.ref_seq                  sat2.NG_048068.1\n",
       "               sul1.ref_seq                   sul1.NG_048081.1\n",
       "               sul2.ref_seq                   sul2.NG_051852.1\n",
       "Length: 155, dtype: object"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NG=referenced.stack()\n",
    "NG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['NG_047213.1',\n",
       " 'NG_056043.1',\n",
       " 'NG_051846.1',\n",
       " 'NG_047430.1',\n",
       " 'NG_050049.1',\n",
       " 'NG_049257.1',\n",
       " 'NG_049830.1',\n",
       " 'NG_050299.1',\n",
       " 'NG_047587.1',\n",
       " 'NG_052052.1',\n",
       " 'NG_047689.1',\n",
       " 'NG_051497.1',\n",
       " 'NG_047985.1',\n",
       " 'NG_048024.1',\n",
       " 'NG_048025.1',\n",
       " 'NG_048042.1',\n",
       " 'NG_048082.1',\n",
       " 'NG_047213.1',\n",
       " 'NG_051844.1',\n",
       " 'NG_052510.1',\n",
       " 'NG_050590.1',\n",
       " 'NG_050336.1',\n",
       " 'NG_052052.1',\n",
       " 'NG_056035.1',\n",
       " 'NG_051497.1',\n",
       " 'NG_050423.1',\n",
       " 'NG_050436.1',\n",
       " 'NG_050543.1',\n",
       " 'NG_047430.1',\n",
       " 'NG_050121.1',\n",
       " 'NG_049762.1',\n",
       " 'NG_050145.1',\n",
       " 'NG_047587.1',\n",
       " 'NG_051699.1',\n",
       " 'NG_056035.1',\n",
       " 'NG_047763.1',\n",
       " 'NG_051497.1',\n",
       " 'NG_050418.1',\n",
       " 'NG_050441.1',\n",
       " 'NG_048042.1',\n",
       " 'NG_048082.1',\n",
       " 'NG_048177.1',\n",
       " 'NG_047213.1',\n",
       " 'NG_047251.1',\n",
       " 'NG_056043.1',\n",
       " 'NG_052310.1',\n",
       " 'NG_052536.1',\n",
       " 'NG_056002.2',\n",
       " 'NG_047466.1',\n",
       " 'NG_047476.1',\n",
       " 'NG_048580.1',\n",
       " 'NG_048834.1',\n",
       " 'NG_050105.1',\n",
       " 'NG_049075.1',\n",
       " 'NG_063835.1',\n",
       " 'NG_048935.1',\n",
       " 'NG_049326.1',\n",
       " 'NG_049392.1',\n",
       " 'NG_049830.1',\n",
       " 'NG_050299.1',\n",
       " 'NG_047559.1',\n",
       " 'NG_047587.1',\n",
       " 'NG_052455.1',\n",
       " 'NG_051436.1',\n",
       " 'NG_047676.1',\n",
       " 'NG_047770.1',\n",
       " 'NG_051497.1',\n",
       " 'NG_064660.1',\n",
       " 'NG_048007.1',\n",
       " 'NG_050424.1',\n",
       " 'NG_050439.1',\n",
       " 'NG_048042.1',\n",
       " 'NG_048068.1',\n",
       " 'NG_048081.1',\n",
       " 'NG_051852.1',\n",
       " 'NG_047213.1',\n",
       " 'NG_056043.1',\n",
       " 'NG_047343.1',\n",
       " 'NG_047430.1',\n",
       " 'NG_050049.1',\n",
       " 'NG_049257.1',\n",
       " 'NG_049830.1',\n",
       " 'NG_050299.1',\n",
       " 'NG_047587.1',\n",
       " 'NG_052052.1',\n",
       " 'NG_047689.1',\n",
       " 'NG_051497.1',\n",
       " 'NG_047985.1',\n",
       " 'NG_048024.1',\n",
       " 'NG_048025.1',\n",
       " 'NG_048042.1',\n",
       " 'NG_048082.1',\n",
       " 'NG_047213.1',\n",
       " 'NG_047251.1',\n",
       " 'NG_051844.1',\n",
       " 'NG_052405.1',\n",
       " 'NG_052437.1',\n",
       " 'NG_051908.1',\n",
       " 'NG_052510.1',\n",
       " 'NG_050590.1',\n",
       " 'NG_050336.1',\n",
       " 'NG_052052.1',\n",
       " 'NG_056035.1',\n",
       " 'NG_051497.1',\n",
       " 'NG_050423.1',\n",
       " 'NG_050436.1',\n",
       " 'NG_050543.1',\n",
       " 'NG_048089.1',\n",
       " 'NG_051852.1',\n",
       " 'NG_047430.1',\n",
       " 'NG_050121.1',\n",
       " 'NG_049762.1',\n",
       " 'NG_050145.1',\n",
       " 'NG_047587.1',\n",
       " 'NG_051699.1',\n",
       " 'NG_056035.1',\n",
       " 'NG_047763.1',\n",
       " 'NG_051497.1',\n",
       " 'NG_050418.1',\n",
       " 'NG_050441.1',\n",
       " 'NG_048042.1',\n",
       " 'NG_048082.1',\n",
       " 'NG_048177.1',\n",
       " 'NG_047213.1',\n",
       " 'NG_047251.1',\n",
       " 'NG_056043.1',\n",
       " 'NG_052310.1',\n",
       " 'NG_052536.1',\n",
       " 'NG_056002.2',\n",
       " 'NG_047466.1',\n",
       " 'NG_047476.1',\n",
       " 'NG_048580.1',\n",
       " 'NG_048834.1',\n",
       " 'NG_050105.1',\n",
       " 'NG_049075.1',\n",
       " 'NG_048935.1',\n",
       " 'NG_049326.1',\n",
       " 'NG_049392.1',\n",
       " 'NG_049830.1',\n",
       " 'NG_050266.1',\n",
       " 'NG_047559.1',\n",
       " 'NG_047587.1',\n",
       " 'NG_052455.1',\n",
       " 'NG_051436.1',\n",
       " 'NG_047676.1',\n",
       " 'NG_047770.1',\n",
       " 'NG_051497.1',\n",
       " 'NG_064660.1',\n",
       " 'NG_048007.1',\n",
       " 'NG_050424.1',\n",
       " 'NG_050439.1',\n",
       " 'NG_048042.1',\n",
       " 'NG_048068.1',\n",
       " 'NG_048081.1',\n",
       " 'NG_051852.1']"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NG_separados=[\".\".join(i.split(\".\")[1:3]) for i in NG.values]\n",
    "NG_separados"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>refseq_nucleotide_accession</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPN12_S2_L001</th>\n",
       "      <th>A7J11_01233+.ref_seq</th>\n",
       "      <td>NG_047213.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aac_6___Ib+.ref_seq</th>\n",
       "      <td>NG_056043.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aadA2+.ref_seq</th>\n",
       "      <td>NG_051846.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aph_3___Ia.ref_seq</th>\n",
       "      <td>NG_047430.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bla-_6.ref_seq</th>\n",
       "      <td>NG_050049.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPNK_S1_L001</th>\n",
       "      <th>oqxB-.ref_seq</th>\n",
       "      <td>NG_050439.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>qacE-.ref_seq</th>\n",
       "      <td>NG_048042.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sat2+.ref_seq</th>\n",
       "      <td>NG_048068.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul1.ref_seq</th>\n",
       "      <td>NG_048081.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul2.ref_seq</th>\n",
       "      <td>NG_051852.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>155 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   refseq_nucleotide_accession\n",
       "samples                                                       \n",
       "KPN12_S2_L001 A7J11_01233+.ref_seq                 NG_047213.1\n",
       "              aac_6___Ib+.ref_seq                  NG_056043.1\n",
       "              aadA2+.ref_seq                       NG_051846.1\n",
       "              aph_3___Ia.ref_seq                   NG_047430.1\n",
       "              bla-_6.ref_seq                       NG_050049.1\n",
       "...                                                        ...\n",
       "KPNK_S1_L001  oqxB-.ref_seq                        NG_050439.1\n",
       "              qacE-.ref_seq                        NG_048042.1\n",
       "              sat2+.ref_seq                        NG_048068.1\n",
       "              sul1.ref_seq                         NG_048081.1\n",
       "              sul2.ref_seq                         NG_051852.1\n",
       "\n",
       "[155 rows x 1 columns]"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NG_arreglados=pd.DataFrame(NG_separados,index=NG.index)\n",
    "NG_arreglados.columns=[\"refseq_nucleotide_accession\"]\n",
    "NG_arreglados"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th>refseq_nucleotide_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>A7J11_01233+.ref_seq</td>\n",
       "      <td>NG_047213.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aac_6___Ib+.ref_seq</td>\n",
       "      <td>NG_056043.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aadA2+.ref_seq</td>\n",
       "      <td>NG_051846.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aph_3___Ia.ref_seq</td>\n",
       "      <td>NG_047430.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>bla-_6.ref_seq</td>\n",
       "      <td>NG_050049.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>oqxB-.ref_seq</td>\n",
       "      <td>NG_050439.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>qacE-.ref_seq</td>\n",
       "      <td>NG_048042.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sat2+.ref_seq</td>\n",
       "      <td>NG_048068.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sul1.ref_seq</td>\n",
       "      <td>NG_048081.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sul2.ref_seq</td>\n",
       "      <td>NG_051852.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>155 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           samples               level_1 refseq_nucleotide_accession\n",
       "0    KPN12_S2_L001  A7J11_01233+.ref_seq                 NG_047213.1\n",
       "1    KPN12_S2_L001   aac_6___Ib+.ref_seq                 NG_056043.1\n",
       "2    KPN12_S2_L001        aadA2+.ref_seq                 NG_051846.1\n",
       "3    KPN12_S2_L001    aph_3___Ia.ref_seq                 NG_047430.1\n",
       "4    KPN12_S2_L001        bla-_6.ref_seq                 NG_050049.1\n",
       "..             ...                   ...                         ...\n",
       "150   KPNK_S1_L001         oqxB-.ref_seq                 NG_050439.1\n",
       "151   KPNK_S1_L001         qacE-.ref_seq                 NG_048042.1\n",
       "152   KPNK_S1_L001         sat2+.ref_seq                 NG_048068.1\n",
       "153   KPNK_S1_L001          sul1.ref_seq                 NG_048081.1\n",
       "154   KPNK_S1_L001          sul2.ref_seq                 NG_051852.1\n",
       "\n",
       "[155 rows x 3 columns]"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NG_arreglados_df = NG_arreglados.reset_index(level=[0,1])\n",
    "NG_arreglados_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>level_1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A7J11_01233+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>aac_6___Ib+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>aadA2+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>aph_3___Ia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bla-_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>oqxB-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>qacE-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>sat2+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>sul1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>sul2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>155 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          level_1\n",
       "0    A7J11_01233+\n",
       "1     aac_6___Ib+\n",
       "2          aadA2+\n",
       "3      aph_3___Ia\n",
       "4          bla-_6\n",
       "..            ...\n",
       "150         oqxB-\n",
       "151         qacE-\n",
       "152         sat2+\n",
       "153          sul1\n",
       "154          sul2\n",
       "\n",
       "[155 rows x 1 columns]"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fixed_column_NG_arreglados_df=NG_arreglados_df.level_1.str.rstrip('.ref_seq')\n",
    "fixed_column_NG_arreglados_df=pd.DataFrame(fixed_column_NG_arreglados_df)\n",
    "fixed_column_NG_arreglados_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th>refseq_nucleotide_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>A7J11_01233+</td>\n",
       "      <td>NG_047213.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aac_6___Ib+</td>\n",
       "      <td>NG_056043.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aadA2+</td>\n",
       "      <td>NG_051846.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>aph_3___Ia</td>\n",
       "      <td>NG_047430.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KPN12_S2_L001</td>\n",
       "      <td>bla-_6</td>\n",
       "      <td>NG_050049.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>oqxB-</td>\n",
       "      <td>NG_050439.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>qacE-</td>\n",
       "      <td>NG_048042.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sat2+</td>\n",
       "      <td>NG_048068.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sul1</td>\n",
       "      <td>NG_048081.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>KPNK_S1_L001</td>\n",
       "      <td>sul2</td>\n",
       "      <td>NG_051852.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>155 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           samples       level_1 refseq_nucleotide_accession\n",
       "0    KPN12_S2_L001  A7J11_01233+                 NG_047213.1\n",
       "1    KPN12_S2_L001   aac_6___Ib+                 NG_056043.1\n",
       "2    KPN12_S2_L001        aadA2+                 NG_051846.1\n",
       "3    KPN12_S2_L001    aph_3___Ia                 NG_047430.1\n",
       "4    KPN12_S2_L001        bla-_6                 NG_050049.1\n",
       "..             ...           ...                         ...\n",
       "150   KPNK_S1_L001         oqxB-                 NG_050439.1\n",
       "151   KPNK_S1_L001         qacE-                 NG_048042.1\n",
       "152   KPNK_S1_L001         sat2+                 NG_048068.1\n",
       "153   KPNK_S1_L001          sul1                 NG_048081.1\n",
       "154   KPNK_S1_L001          sul2                 NG_051852.1\n",
       "\n",
       "[155 rows x 3 columns]"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NG_arreglados_df['level_1']=fixed_column_NG_arreglados_df['level_1']\n",
    "NG_arreglados_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>refseq_nucleotide_accession</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPN12_S2_L001</th>\n",
       "      <th>A7J11_01233+</th>\n",
       "      <td>NG_047213.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aac_6___Ib+</th>\n",
       "      <td>NG_056043.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aadA2+</th>\n",
       "      <td>NG_051846.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aph_3___Ia</th>\n",
       "      <td>NG_047430.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bla-_6</th>\n",
       "      <td>NG_050049.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPNK_S1_L001</th>\n",
       "      <th>oqxB-</th>\n",
       "      <td>NG_050439.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>qacE-</th>\n",
       "      <td>NG_048042.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sat2+</th>\n",
       "      <td>NG_048068.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul1</th>\n",
       "      <td>NG_048081.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul2</th>\n",
       "      <td>NG_051852.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>155 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           refseq_nucleotide_accession\n",
       "samples       level_1                                 \n",
       "KPN12_S2_L001 A7J11_01233+                 NG_047213.1\n",
       "              aac_6___Ib+                  NG_056043.1\n",
       "              aadA2+                       NG_051846.1\n",
       "              aph_3___Ia                   NG_047430.1\n",
       "              bla-_6                       NG_050049.1\n",
       "...                                                ...\n",
       "KPNK_S1_L001  oqxB-                        NG_050439.1\n",
       "              qacE-                        NG_048042.1\n",
       "              sat2+                        NG_048068.1\n",
       "              sul1                         NG_048081.1\n",
       "              sul2                         NG_051852.1\n",
       "\n",
       "[155 rows x 1 columns]"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#####----------------------------------> data frame 3 ---> NG arreglados!!\n",
    "NG_fixed=NG_arreglados_df.set_index(['samples', 'level_1'])\n",
    "NG_fixed\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>mutation_result</th>\n",
       "      <th>gene family</th>\n",
       "      <th>refseq nucleotide</th>\n",
       "      <th>class</th>\n",
       "      <th>subclass</th>\n",
       "      <th>refseq protein</th>\n",
       "      <th>genbank protein</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPN12_S2_L001</th>\n",
       "      <th>acrR_F172S</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_F197I</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_G164A</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_K201M</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_L195V</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPNK_S1_L001</th>\n",
       "      <th>ompK36_L59V</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK36_N218H</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK36_N49S</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK37_I128M</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK37_I70M</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>155 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           mutation_result gene family refseq nucleotide  \\\n",
       "samples       level_1                                                      \n",
       "KPN12_S2_L001 acrR_F172S               yes         NaN               NaN   \n",
       "              acrR_F197I               yes         NaN               NaN   \n",
       "              acrR_G164A               yes         NaN               NaN   \n",
       "              acrR_K201M               yes         NaN               NaN   \n",
       "              acrR_L195V               yes         NaN               NaN   \n",
       "...                                    ...         ...               ...   \n",
       "KPNK_S1_L001  ompK36_L59V              yes         NaN               NaN   \n",
       "              ompK36_N218H             yes         NaN               NaN   \n",
       "              ompK36_N49S              yes         NaN               NaN   \n",
       "              ompK37_I128M             yes         NaN               NaN   \n",
       "              ompK37_I70M              yes         NaN               NaN   \n",
       "\n",
       "                           class subclass refseq protein genbank protein  \n",
       "samples       level_1                                                     \n",
       "KPN12_S2_L001 acrR_F172S     NaN      NaN            NaN             NaN  \n",
       "              acrR_F197I     NaN      NaN            NaN             NaN  \n",
       "              acrR_G164A     NaN      NaN            NaN             NaN  \n",
       "              acrR_K201M     NaN      NaN            NaN             NaN  \n",
       "              acrR_L195V     NaN      NaN            NaN             NaN  \n",
       "...                          ...      ...            ...             ...  \n",
       "KPNK_S1_L001  ompK36_L59V    NaN      NaN            NaN             NaN  \n",
       "              ompK36_N218H   NaN      NaN            NaN             NaN  \n",
       "              ompK36_N49S    NaN      NaN            NaN             NaN  \n",
       "              ompK37_I128M   NaN      NaN            NaN             NaN  \n",
       "              ompK37_I70M    NaN      NaN            NaN             NaN  \n",
       "\n",
       "[155 rows x 7 columns]"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### Union entre las mutaciones y la base de datos:\n",
    "mutaciones_finales=mutation_result_df.join(database_anthony.set_index(\"#allele\"),on=\"level_1\")\n",
    "mutaciones_finales\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#allele</th>\n",
       "      <th>gene family</th>\n",
       "      <th>refseq nucleotide</th>\n",
       "      <th>class</th>\n",
       "      <th>subclass</th>\n",
       "      <th>refseq protein</th>\n",
       "      <th>genbank protein</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(3)-IId</td>\n",
       "      <td>NG_047251.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>GENTAMICIN</td>\n",
       "      <td>WP_000557454.1</td>\n",
       "      <td>AAN87703.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(6')-Ib</td>\n",
       "      <td>NG_056043.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>AMIKACIN/KANAMYCIN/TOBRAMYCIN</td>\n",
       "      <td>WP_063840280.1</td>\n",
       "      <td>KSL18989.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>aac(6')-Ib4</td>\n",
       "      <td>aac(6')-Ib</td>\n",
       "      <td>NG_051844.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>GENTAMICIN</td>\n",
       "      <td>WP_003159191.1</td>\n",
       "      <td>AAF72941.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aadA1</td>\n",
       "      <td>NG_052536.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>STREPTOMYCIN</td>\n",
       "      <td>WP_046622360.1</td>\n",
       "      <td>KKJ21274.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>NaN</td>\n",
       "      <td>aadA13</td>\n",
       "      <td>NG_052310.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>STREPTOMYCIN</td>\n",
       "      <td>WP_071846269.1</td>\n",
       "      <td>AGY30809.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5892</th>\n",
       "      <td>NaN</td>\n",
       "      <td>sul1</td>\n",
       "      <td>NG_048081.1</td>\n",
       "      <td>SULFONAMIDE</td>\n",
       "      <td>SULFONAMIDE</td>\n",
       "      <td>WP_000259032.1</td>\n",
       "      <td>CAA33123.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5902</th>\n",
       "      <td>NaN</td>\n",
       "      <td>sul2</td>\n",
       "      <td>NG_051852.1</td>\n",
       "      <td>SULFONAMIDE</td>\n",
       "      <td>SULFONAMIDE</td>\n",
       "      <td>WP_001043260.1</td>\n",
       "      <td>AAK64592.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6025</th>\n",
       "      <td>NaN</td>\n",
       "      <td>tet(C)</td>\n",
       "      <td>NG_048177.1</td>\n",
       "      <td>TETRACYCLINE</td>\n",
       "      <td>TETRACYCLINE</td>\n",
       "      <td>WP_001297013.1</td>\n",
       "      <td>AAK97755.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6512</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NG_052052.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>GENTAMICIN</td>\n",
       "      <td>WP_071593228.1</td>\n",
       "      <td>BAK79091.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6513</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NG_047213.1</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>GENTAMICIN</td>\n",
       "      <td>WP_063838853.1</td>\n",
       "      <td>CAE48335.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          #allele gene family refseq nucleotide           class  \\\n",
       "16            NaN  aac(3)-IId       NG_047251.1  AMINOGLYCOSIDE   \n",
       "241           NaN  aac(6')-Ib       NG_056043.1  AMINOGLYCOSIDE   \n",
       "279   aac(6')-Ib4  aac(6')-Ib       NG_051844.1  AMINOGLYCOSIDE   \n",
       "377           NaN       aadA1       NG_052536.1  AMINOGLYCOSIDE   \n",
       "394           NaN      aadA13       NG_052310.1  AMINOGLYCOSIDE   \n",
       "...           ...         ...               ...             ...   \n",
       "5892          NaN        sul1       NG_048081.1     SULFONAMIDE   \n",
       "5902          NaN        sul2       NG_051852.1     SULFONAMIDE   \n",
       "6025          NaN      tet(C)       NG_048177.1    TETRACYCLINE   \n",
       "6512          NaN         NaN       NG_052052.1  AMINOGLYCOSIDE   \n",
       "6513          NaN         NaN       NG_047213.1  AMINOGLYCOSIDE   \n",
       "\n",
       "                           subclass  refseq protein genbank protein  \n",
       "16                       GENTAMICIN  WP_000557454.1      AAN87703.1  \n",
       "241   AMIKACIN/KANAMYCIN/TOBRAMYCIN  WP_063840280.1      KSL18989.2  \n",
       "279                      GENTAMICIN  WP_003159191.1      AAF72941.1  \n",
       "377                    STREPTOMYCIN  WP_046622360.1      KKJ21274.1  \n",
       "394                    STREPTOMYCIN  WP_071846269.1      AGY30809.1  \n",
       "...                             ...             ...             ...  \n",
       "5892                    SULFONAMIDE  WP_000259032.1      CAA33123.1  \n",
       "5902                    SULFONAMIDE  WP_001043260.1      AAK64592.1  \n",
       "6025                   TETRACYCLINE  WP_001297013.1      AAK97755.1  \n",
       "6512                     GENTAMICIN  WP_071593228.1      BAK79091.1  \n",
       "6513                     GENTAMICIN  WP_063838853.1      CAE48335.2  \n",
       "\n",
       "[65 rows x 7 columns]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tabla_final_NCBI=database_anthony[database_anthony['refseq nucleotide'].isin(NG_separados)]\n",
    "tabla_final_NCBI.to_csv(\"tabla_final_NCBI.csv\")\n",
    "tabla_final_NCBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>refseq_nucleotide_accession</th>\n",
       "      <th>#allele</th>\n",
       "      <th>gene family</th>\n",
       "      <th>class</th>\n",
       "      <th>subclass</th>\n",
       "      <th>refseq protein</th>\n",
       "      <th>genbank protein</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPN12_S2_L001</th>\n",
       "      <th>A7J11_01233+</th>\n",
       "      <td>NG_047213.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>GENTAMICIN</td>\n",
       "      <td>WP_063838853.1</td>\n",
       "      <td>CAE48335.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aac_6___Ib+</th>\n",
       "      <td>NG_056043.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(6')-Ib</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>AMIKACIN/KANAMYCIN/TOBRAMYCIN</td>\n",
       "      <td>WP_063840280.1</td>\n",
       "      <td>KSL18989.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aadA2+</th>\n",
       "      <td>NG_051846.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aadA2</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>STREPTOMYCIN</td>\n",
       "      <td>WP_001261740.1</td>\n",
       "      <td>AAF87686.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aph_3___Ia</th>\n",
       "      <td>NG_047430.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aph(3')-Ia</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>KANAMYCIN</td>\n",
       "      <td>WP_000018329.1</td>\n",
       "      <td>CAA23656.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bla-_6</th>\n",
       "      <td>NG_050049.1</td>\n",
       "      <td>blaSHV-182</td>\n",
       "      <td>blaSHV</td>\n",
       "      <td>BETA-LACTAM</td>\n",
       "      <td>BETA-LACTAM</td>\n",
       "      <td>WP_063864660.1</td>\n",
       "      <td>AJO16042.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPNK_S1_L001</th>\n",
       "      <th>oqxB-</th>\n",
       "      <td>NG_050439.1</td>\n",
       "      <td>oqxB20</td>\n",
       "      <td>oqxB</td>\n",
       "      <td>PHENICOL/QUINOLONE</td>\n",
       "      <td>PHENICOL/QUINOLONE</td>\n",
       "      <td>WP_021312368.1</td>\n",
       "      <td>AGZ20241.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>qacE-</th>\n",
       "      <td>NG_048042.1</td>\n",
       "      <td>qacEdelta1</td>\n",
       "      <td>qacE</td>\n",
       "      <td>QUATERNARY AMMONIUM</td>\n",
       "      <td>QUATERNARY AMMONIUM</td>\n",
       "      <td>WP_000679427.1</td>\n",
       "      <td>AAC44316.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sat2+</th>\n",
       "      <td>NG_048068.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sat2</td>\n",
       "      <td>STREPTOTHRICIN</td>\n",
       "      <td>STREPTOTHRICIN</td>\n",
       "      <td>WP_000704156.1</td>\n",
       "      <td>CAA34124.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul1</th>\n",
       "      <td>NG_048081.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sul1</td>\n",
       "      <td>SULFONAMIDE</td>\n",
       "      <td>SULFONAMIDE</td>\n",
       "      <td>WP_000259032.1</td>\n",
       "      <td>CAA33123.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul2</th>\n",
       "      <td>NG_051852.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sul2</td>\n",
       "      <td>SULFONAMIDE</td>\n",
       "      <td>SULFONAMIDE</td>\n",
       "      <td>WP_001043260.1</td>\n",
       "      <td>AAK64592.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>155 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           refseq_nucleotide_accession     #allele  \\\n",
       "samples       level_1                                                \n",
       "KPN12_S2_L001 A7J11_01233+                 NG_047213.1         NaN   \n",
       "              aac_6___Ib+                  NG_056043.1         NaN   \n",
       "              aadA2+                       NG_051846.1         NaN   \n",
       "              aph_3___Ia                   NG_047430.1         NaN   \n",
       "              bla-_6                       NG_050049.1  blaSHV-182   \n",
       "...                                                ...         ...   \n",
       "KPNK_S1_L001  oqxB-                        NG_050439.1      oqxB20   \n",
       "              qacE-                        NG_048042.1  qacEdelta1   \n",
       "              sat2+                        NG_048068.1         NaN   \n",
       "              sul1                         NG_048081.1         NaN   \n",
       "              sul2                         NG_051852.1         NaN   \n",
       "\n",
       "                           gene family                class  \\\n",
       "samples       level_1                                         \n",
       "KPN12_S2_L001 A7J11_01233+         NaN       AMINOGLYCOSIDE   \n",
       "              aac_6___Ib+   aac(6')-Ib       AMINOGLYCOSIDE   \n",
       "              aadA2+             aadA2       AMINOGLYCOSIDE   \n",
       "              aph_3___Ia    aph(3')-Ia       AMINOGLYCOSIDE   \n",
       "              bla-_6            blaSHV          BETA-LACTAM   \n",
       "...                                ...                  ...   \n",
       "KPNK_S1_L001  oqxB-               oqxB   PHENICOL/QUINOLONE   \n",
       "              qacE-               qacE  QUATERNARY AMMONIUM   \n",
       "              sat2+               sat2       STREPTOTHRICIN   \n",
       "              sul1                sul1          SULFONAMIDE   \n",
       "              sul2                sul2          SULFONAMIDE   \n",
       "\n",
       "                                                 subclass  refseq protein  \\\n",
       "samples       level_1                                                       \n",
       "KPN12_S2_L001 A7J11_01233+                     GENTAMICIN  WP_063838853.1   \n",
       "              aac_6___Ib+   AMIKACIN/KANAMYCIN/TOBRAMYCIN  WP_063840280.1   \n",
       "              aadA2+                         STREPTOMYCIN  WP_001261740.1   \n",
       "              aph_3___Ia                        KANAMYCIN  WP_000018329.1   \n",
       "              bla-_6                          BETA-LACTAM  WP_063864660.1   \n",
       "...                                                   ...             ...   \n",
       "KPNK_S1_L001  oqxB-                    PHENICOL/QUINOLONE  WP_021312368.1   \n",
       "              qacE-                   QUATERNARY AMMONIUM  WP_000679427.1   \n",
       "              sat2+                        STREPTOTHRICIN  WP_000704156.1   \n",
       "              sul1                            SULFONAMIDE  WP_000259032.1   \n",
       "              sul2                            SULFONAMIDE  WP_001043260.1   \n",
       "\n",
       "                           genbank protein  \n",
       "samples       level_1                       \n",
       "KPN12_S2_L001 A7J11_01233+      CAE48335.2  \n",
       "              aac_6___Ib+       KSL18989.2  \n",
       "              aadA2+            AAF87686.1  \n",
       "              aph_3___Ia        CAA23656.1  \n",
       "              bla-_6            AJO16042.1  \n",
       "...                                    ...  \n",
       "KPNK_S1_L001  oqxB-             AGZ20241.1  \n",
       "              qacE-             AAC44316.1  \n",
       "              sat2+             CAA34124.1  \n",
       "              sul1              CAA33123.1  \n",
       "              sul2              AAK64592.1  \n",
       "\n",
       "[155 rows x 7 columns]"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resultado_final=NG_fixed.join(tabla_final_NCBI.set_index(\"refseq nucleotide\"),on=\"refseq_nucleotide_accession\")\n",
    "resultado_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>refseq_nucleotide_accession</th>\n",
       "      <th>#allele</th>\n",
       "      <th>gene family</th>\n",
       "      <th>class</th>\n",
       "      <th>subclass</th>\n",
       "      <th>refseq protein</th>\n",
       "      <th>genbank protein</th>\n",
       "      <th>yes/yes_nonunique</th>\n",
       "      <th>coverage</th>\n",
       "      <th>%_ID</th>\n",
       "      <th>mutation_result</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPN12_S2_L001</th>\n",
       "      <th>A7J11_01233+</th>\n",
       "      <td>NG_047213.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>GENTAMICIN</td>\n",
       "      <td>WP_063838853.1</td>\n",
       "      <td>CAE48335.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>51.8</td>\n",
       "      <td>99.83</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aac_6___Ib+</th>\n",
       "      <td>NG_056043.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aac(6')-Ib</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>AMIKACIN/KANAMYCIN/TOBRAMYCIN</td>\n",
       "      <td>WP_063840280.1</td>\n",
       "      <td>KSL18989.2</td>\n",
       "      <td>yes</td>\n",
       "      <td>46.5</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aadA2+</th>\n",
       "      <td>NG_051846.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>aadA2</td>\n",
       "      <td>AMINOGLYCOSIDE</td>\n",
       "      <td>STREPTOMYCIN</td>\n",
       "      <td>WP_001261740.1</td>\n",
       "      <td>AAF87686.1</td>\n",
       "      <td>yes_nonunique</td>\n",
       "      <td>36.5</td>\n",
       "      <td>99.87</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_F172S</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPNK_S1_L001</th>\n",
       "      <th>oqxB-</th>\n",
       "      <td>NG_050439.1</td>\n",
       "      <td>oqxB20</td>\n",
       "      <td>oqxB</td>\n",
       "      <td>PHENICOL/QUINOLONE</td>\n",
       "      <td>PHENICOL/QUINOLONE</td>\n",
       "      <td>WP_021312368.1</td>\n",
       "      <td>AGZ20241.1</td>\n",
       "      <td>yes</td>\n",
       "      <td>39.2</td>\n",
       "      <td>99.90</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>qacE-</th>\n",
       "      <td>NG_048042.1</td>\n",
       "      <td>qacEdelta1</td>\n",
       "      <td>qacE</td>\n",
       "      <td>QUATERNARY AMMONIUM</td>\n",
       "      <td>QUATERNARY AMMONIUM</td>\n",
       "      <td>WP_000679427.1</td>\n",
       "      <td>AAC44316.1</td>\n",
       "      <td>yes</td>\n",
       "      <td>94.7</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sat2+</th>\n",
       "      <td>NG_048068.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sat2</td>\n",
       "      <td>STREPTOTHRICIN</td>\n",
       "      <td>STREPTOTHRICIN</td>\n",
       "      <td>WP_000704156.1</td>\n",
       "      <td>CAA34124.1</td>\n",
       "      <td>yes</td>\n",
       "      <td>29.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul1</th>\n",
       "      <td>NG_048081.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sul1</td>\n",
       "      <td>SULFONAMIDE</td>\n",
       "      <td>SULFONAMIDE</td>\n",
       "      <td>WP_000259032.1</td>\n",
       "      <td>CAA33123.1</td>\n",
       "      <td>yes</td>\n",
       "      <td>166.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sul2</th>\n",
       "      <td>NG_051852.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sul2</td>\n",
       "      <td>SULFONAMIDE</td>\n",
       "      <td>SULFONAMIDE</td>\n",
       "      <td>WP_001043260.1</td>\n",
       "      <td>AAK64592.1</td>\n",
       "      <td>yes</td>\n",
       "      <td>139.6</td>\n",
       "      <td>100.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>347 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           refseq_nucleotide_accession     #allele  \\\n",
       "samples       level_1                                                \n",
       "KPN12_S2_L001 A7J11_01233+                 NG_047213.1         NaN   \n",
       "              aac_6___Ib+                  NG_056043.1         NaN   \n",
       "              aadA2+                       NG_051846.1         NaN   \n",
       "              acrR                                 NaN         NaN   \n",
       "              acrR_F172S                           NaN         NaN   \n",
       "...                                                ...         ...   \n",
       "KPNK_S1_L001  oqxB-                        NG_050439.1      oqxB20   \n",
       "              qacE-                        NG_048042.1  qacEdelta1   \n",
       "              sat2+                        NG_048068.1         NaN   \n",
       "              sul1                         NG_048081.1         NaN   \n",
       "              sul2                         NG_051852.1         NaN   \n",
       "\n",
       "                           gene family                class  \\\n",
       "samples       level_1                                         \n",
       "KPN12_S2_L001 A7J11_01233+         NaN       AMINOGLYCOSIDE   \n",
       "              aac_6___Ib+   aac(6')-Ib       AMINOGLYCOSIDE   \n",
       "              aadA2+             aadA2       AMINOGLYCOSIDE   \n",
       "              acrR                 NaN                  NaN   \n",
       "              acrR_F172S           NaN                  NaN   \n",
       "...                                ...                  ...   \n",
       "KPNK_S1_L001  oqxB-               oqxB   PHENICOL/QUINOLONE   \n",
       "              qacE-               qacE  QUATERNARY AMMONIUM   \n",
       "              sat2+               sat2       STREPTOTHRICIN   \n",
       "              sul1                sul1          SULFONAMIDE   \n",
       "              sul2                sul2          SULFONAMIDE   \n",
       "\n",
       "                                                 subclass  refseq protein  \\\n",
       "samples       level_1                                                       \n",
       "KPN12_S2_L001 A7J11_01233+                     GENTAMICIN  WP_063838853.1   \n",
       "              aac_6___Ib+   AMIKACIN/KANAMYCIN/TOBRAMYCIN  WP_063840280.1   \n",
       "              aadA2+                         STREPTOMYCIN  WP_001261740.1   \n",
       "              acrR                                    NaN             NaN   \n",
       "              acrR_F172S                              NaN             NaN   \n",
       "...                                                   ...             ...   \n",
       "KPNK_S1_L001  oqxB-                    PHENICOL/QUINOLONE  WP_021312368.1   \n",
       "              qacE-                   QUATERNARY AMMONIUM  WP_000679427.1   \n",
       "              sat2+                        STREPTOTHRICIN  WP_000704156.1   \n",
       "              sul1                            SULFONAMIDE  WP_000259032.1   \n",
       "              sul2                            SULFONAMIDE  WP_001043260.1   \n",
       "\n",
       "                           genbank protein yes/yes_nonunique  coverage  \\\n",
       "samples       level_1                                                    \n",
       "KPN12_S2_L001 A7J11_01233+      CAE48335.2               NaN      51.8   \n",
       "              aac_6___Ib+       KSL18989.2               yes      46.5   \n",
       "              aadA2+            AAF87686.1     yes_nonunique      36.5   \n",
       "              acrR                     NaN               yes       NaN   \n",
       "              acrR_F172S               NaN               NaN       NaN   \n",
       "...                                    ...               ...       ...   \n",
       "KPNK_S1_L001  oqxB-             AGZ20241.1               yes      39.2   \n",
       "              qacE-             AAC44316.1               yes      94.7   \n",
       "              sat2+             CAA34124.1               yes      29.3   \n",
       "              sul1              CAA33123.1               yes     166.3   \n",
       "              sul2              AAK64592.1               yes     139.6   \n",
       "\n",
       "                              %_ID mutation_result  \n",
       "samples       level_1                               \n",
       "KPN12_S2_L001 A7J11_01233+   99.83             NaN  \n",
       "              aac_6___Ib+   100.00             NaN  \n",
       "              aadA2+         99.87             NaN  \n",
       "              acrR             NaN             NaN  \n",
       "              acrR_F172S       NaN             yes  \n",
       "...                            ...             ...  \n",
       "KPNK_S1_L001  oqxB-          99.90             NaN  \n",
       "              qacE-         100.00             NaN  \n",
       "              sat2+         100.00             NaN  \n",
       "              sul1          100.00             NaN  \n",
       "              sul2          100.00             NaN  \n",
       "\n",
       "[347 rows x 11 columns]"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resultado_integrated= pd.concat([resultado_final, yeah_column_arreglada], axis=1)\n",
    "resultado_integrated=pd.concat([resultado_integrated, final_coverage_fixed], axis=1)\n",
    "resultado_integrated=pd.concat([resultado_integrated, gene_identification_fixed], axis=1)\n",
    "resultado_integrated=pd.concat([resultado_integrated, mutation_result_df], axis=1)\n",
    "resultado_integrated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "resultado_integrated.to_csv(\"Klebsiella_resultado_integrated.csv\")####---> final output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "mutaciones_finales.to_csv(\"Klebsiella_resultado_mutaciones_integrated.csv\")####---> final output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>mutation_result</th>\n",
       "      <th>gene family</th>\n",
       "      <th>refseq nucleotide</th>\n",
       "      <th>class</th>\n",
       "      <th>subclass</th>\n",
       "      <th>refseq protein</th>\n",
       "      <th>genbank protein</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samples</th>\n",
       "      <th>level_1</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPN12_S2_L001</th>\n",
       "      <th>acrR_F172S</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_F197I</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_G164A</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_K201M</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acrR_L195V</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">KPNK_S1_L001</th>\n",
       "      <th>ompK36_L59V</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK36_N218H</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK36_N49S</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK37_I128M</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ompK37_I70M</th>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>155 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           mutation_result gene family refseq nucleotide  \\\n",
       "samples       level_1                                                      \n",
       "KPN12_S2_L001 acrR_F172S               yes         NaN               NaN   \n",
       "              acrR_F197I               yes         NaN               NaN   \n",
       "              acrR_G164A               yes         NaN               NaN   \n",
       "              acrR_K201M               yes         NaN               NaN   \n",
       "              acrR_L195V               yes         NaN               NaN   \n",
       "...                                    ...         ...               ...   \n",
       "KPNK_S1_L001  ompK36_L59V              yes         NaN               NaN   \n",
       "              ompK36_N218H             yes         NaN               NaN   \n",
       "              ompK36_N49S              yes         NaN               NaN   \n",
       "              ompK37_I128M             yes         NaN               NaN   \n",
       "              ompK37_I70M              yes         NaN               NaN   \n",
       "\n",
       "                           class subclass refseq protein genbank protein  \n",
       "samples       level_1                                                     \n",
       "KPN12_S2_L001 acrR_F172S     NaN      NaN            NaN             NaN  \n",
       "              acrR_F197I     NaN      NaN            NaN             NaN  \n",
       "              acrR_G164A     NaN      NaN            NaN             NaN  \n",
       "              acrR_K201M     NaN      NaN            NaN             NaN  \n",
       "              acrR_L195V     NaN      NaN            NaN             NaN  \n",
       "...                          ...      ...            ...             ...  \n",
       "KPNK_S1_L001  ompK36_L59V    NaN      NaN            NaN             NaN  \n",
       "              ompK36_N218H   NaN      NaN            NaN             NaN  \n",
       "              ompK36_N49S    NaN      NaN            NaN             NaN  \n",
       "              ompK37_I128M   NaN      NaN            NaN             NaN  \n",
       "              ompK37_I70M    NaN      NaN            NaN             NaN  \n",
       "\n",
       "[155 rows x 7 columns]"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutaciones_finales\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
